---
title: "Annotating"
output: rmarkdown::html_vignette
description: |
  tidyproteomics: simplified data object and analysis functions
vignette: >
  %\VignetteIndexEntry{Annotating}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---

```{r, echo = FALSE}
knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
set.seed(1041)
options(dplyr.print_max = 10)
```

# Annotating data

As a part of the Tidyproteomics workflow, we need to update the
`gene_name` and `description` annotated terms in the data imported from
FragPipe. To accomplish this, we will use information from a parsed
FASTA file. The parsed FASTA file should contain three columns:
`protein` identifier, `term`, and `annotation`.

Updating the `gene_name` and `description` annotated terms will allow
for easier interpretation and analysis of the data. The `gene_name` and
`description` attributes provide crucial information about the protein,
such as its function and biological role. By updating these attributes
with information from the FASTA file, we can ensure that our data is
accurate and informative.

The process of updating the `gene_name` and `description` annotated
terms is simple. First, we will parse the FASTA file to extract the
necessary information. Then, we will use this information to update the
corresponding attributes in the imported data from FragPipe.

It is important to note that the parsed FASTA file must contain accurate
and up-to-date information. If the information in the FASTA file is
outdated or incorrect, the updated `gene_name` and `description`
attributes will also be incorrect. Therefore, it is essential to verify
the accuracy of the FASTA file before using it to update the imported
data from FragPipe.

```{r, eval=FALSE}
library(tidyverse)
library(tidyproteomics)

# download the data
url <- "https://ftp.ebi.ac.uk/pride-archive/2016/06/PXD004163/Yan_miR_Protein_table.flatprottable.txt"
download.file(url, destfile = "./data/combined_protein.tsv", method = "auto")

# import the data
data_prot <- "./data/combined_protein.tsv" %>% import('FragPipe', 'proteins')
```

```{r, echo=FALSE, warning=FALSE, message=FALSE}
library(tidyverse)
library(tidyproteomics)

data_prot <- "~/Local/data/tidyproteomics/FragPipe_19.1/combined_protein.tsv" %>%
  import('FragPipe', 'proteins')

```

## From a FASTA File

Read in a FASTA file using some "un-exposed" methods in the
`tidyproteomics` package.

```{r, message=FALSE}
data_fasta <- "~/Local/data/fasta/uniprot_human-20398_20220920.fasta" %>% 
  tidyproteomics:::fasta_parse(as = "data.frame") %>%
  select(protein = accession, gene_name, description) %>%
  pivot_longer(
    cols = c('gene_name', 'description'),
    names_to = 'term',
    values_to = 'annotation'
  )

data_fasta %>% filter(protein %in% c('P68431', 'P62805'))

```

Notice the `gene_name` annotations are different from the FASTA than
from the FragPipe outoput.

```{r}
data_prot$annotations %>% filter(protein %in% c('P68431', 'P62805'))

```

We can **merge** the annotations,

```{r}
data_new_merged <- data_prot %>% annotate(data_fasta, duplicates = 'merge')

data_new_merged$annotations %>% filter(protein %in% c('P68431', 'P62805'))

```

Or we can **replace** the annotations,

```{r}
data_new_replaced <- data_prot %>% annotate(data_fasta, duplicates = 'replace')

data_new_replaced$annotations %>% filter(protein %in% c('P68431', 'P62805'))

```

## GO Annotations

To obtain GO annotations, you can visit UniProt's website and search for
the proteins of interest, such as human proteins. Once you have found
the proteins, you will need to select the "Customize columns" option to
access several options, including Gene Ontology.

After selecting Gene Ontology, you will need to choose the desired
values, such as molecular function, by clicking on them. Once you have
selected your desired values, click on the "Save" button to save your
changes.

![*Figure 1 - UniProt web search for human
proteins*](images/uniprot_search_01.png){width="100%"}

Finally, you can download the table as a TSV file by clicking on the
"Download" button. This file will contain all the information you need
about your selected proteins, making it easier to analyze and interpret
the data.

Now that you know how to obtain GO annotations, you can use this
information to enhance your research and analysis. UniProt's website is
a valuable resource for obtaining information about proteins, and the
ability to customize columns and select desired values makes it even
more useful for researchers and scientists.

![*Figure 2 - UniProt show table layout with Go
annotations*](images/uniprot_search_02.png){width="100%"}

Read in the TSV file from the downloaded UniProt table.

```{r, message=FALSE}
data_go <- "~/Local/data/uniprotkb/uniprotkb_human_AND_reviewed_true_AND_m_2023_10_10.tsv" %>% read_tsv()

data_go
```

We just need to tidy up that data a bit and get it into the format
needed for attaching the annotations.

```{r}
data_go <- data_go %>%
  select(protein = Entry,
         molecular_function = `Gene Ontology (molecular function)`) %>%
  # separate the GO terms so we get 1/row
  separate_rows(molecular_function, sep="\\;\\s") %>%
  # remove the [GO:accession]
  mutate(molecular_function = sub("\\s\\[.+", "", molecular_function)) %>%
  # pivot to the needed format
  pivot_longer(molecular_function,
               names_to = 'term',
               values_to = 'annotation')

data_go
```

Looks great!

```{r}
data_new_go <- data_prot %>% annotate(data_go)

data_new_go$annotations %>% filter(protein %in% c('P68431', 'P62805'))
```

Take it for a test drive by subsetting the data based on a specific
annotation term.

```{r, message=FALSE}
data_new_go %>% 
  subset(molecular_function == 'structural constituent of chromatin')

```

An enrichment plot for "ion binding" for the annotation `molecular_function`.

```{r, message=FALSE}
data_new_go %>% 
  subset(molecular_function %like% 'ion binding') %>% 
  expression(knockdown/control) %>% 
  enrichment(knockdown/control, 
             .terms = 'molecular_function',
             .method = 'wilcoxon') %>%
  plot_enrichment(
    knockdown/control, 
    .term = 'molecular_function',
    significance_max = 1
  ) 

```
