---
title: "Expression"
output: rmarkdown::html_vignette
description: |
tidyproteomics: simplified data object and analysis functions
vignette: >
  %\VignetteIndexEntry{Expression}
  %\VignetteEncoding{UTF-8}
  %\VignetteEngine{knitr::rmarkdown}
editor_options: 
  markdown: 
    wrap: 72
---

```{r, echo = FALSE}
knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
set.seed(1041)
options(dplyr.print_max = 10)
```

### Analysis of Two Samples

One of the most fundamental analyses for quantitative proteomics is the
estimation of univariate differences between two samples. Tidyproteomics
accommodates two sample analysis using the `expression()` function, the
results of which can be visualized using the `plot_volcano()` and
`plot_proportion()` visualization functions as shown below. The
expression analysis can be accomplished thought the R *limma* package,
or using any two-sample statistical comparison, the default method
employs the two-sided T-test. Note, if no values pass the cutoff it will
automatically choose the top proteins to display.

It should be noted that this package does not incorporate some of the
more recent methods described in literature. These methods could be
incorporated in future revisions, or through a user provided function.

#### Available Methods

+---------------+---------------------------------------------------------------------------------------------------+
| t.test        | Johnston, L. W. "Student's t-Test." *Journal of                                                   |
|               | Quality Technology* 2.4 (1970): 243-245.                                                          |
|               |                                                                                                   |
|               | [R::stats::t.test](https://stat.eth%%2020z%20.ch/R-manual/R-devel/library/stats/html/t.test.html) |
+---------------+---------------------------------------------------------------------------------------------------+
| limma         | Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W,                                                 |
|               | Smyth GK (2015). "limma powers differential                                                       |
| (Empirical    | expression analyses for RNA-sequencing and                                                        |
| Bayes)        | microarray studies." *Nucleic Acids Research*,                                                    |
|               | **43**(7), e47.                                                                                   |
|               |                                                                                                   |
|               | [Bioconductor::limma](http://bi%%2020o%20conductor.org/packages/release/bioc/html/limma.html)     |
+---------------+---------------------------------------------------------------------------------------------------+

#### Alternative Methods

+---------------+--------------------------------------------------+
| MSstats       | Choi, Meena, Ching-Yun Chang, Timothy Clough,    |
|               | Daniel Broudy, Trevor Killeen, Brendan MacLean,  |
|               | and Olga Vitek. 2014. "MSstats: An R Package for |
|               | Statistical Analysis of Quantitative Mass        |
|               | Spectrometry-Based Proteomic Experiments."       |
|               | *Bioinformatics* 30 (17): 2524--26.              |
|               |                                                  |
|               | [msstats.org](msstats.org)                       |
+---------------+--------------------------------------------------+
| DEqMS         | Zhu, Yafeng, Lukas M. Orre, Yan Zhou Tran,       |
|               | Georgios Mermelekas, Henrik J. Johansson, Alina  |
|               | Malyutina, Simon Anders, and Janne Lehti√∂. 2020. |
|               | "DEqMS: A Method for Accurate Variance           |
|               | Estimation in Differential Protein Expression    |
|               | Analysis." *Molecular & Cellular Proteomics:     |
|               | MCP* 19 (6): 1047--57.                           |
|               |                                                  |
|               | [Bioconductor:                                   |
|               | DEqMS](https://w%20ww.bi%20oco%20nd              |
|               | uctor.org/packages/release/bioc/html/DEqMS.html) |
+---------------+--------------------------------------------------+
| MS -EmpiRe    | Ammar, Constantin, Markus Gruber, Gergely Csaba, |
|               | and Ralf Zimmer. 2019. "MS-EmpiRe Utilizes       |
|               | Peptide-Level Noise Distributions for            |
|               | Ultra-Sensitive Detection of Differentially      |
|               | Expressed Proteins." *Molecular & Cellular       |
|               | Proteomics: MCP* 18 (9): 1880--92.               |
|               |                                                  |
|               | [GitHub::MS-EmpiRe](https://github.com/zimmerlab/MS-EmpiRe) |
+---------------+--------------------------------------------------+
| meqrob        | Sticker, Adriaan, Ludger Goeminne, Lennart       |
|               | Martens, and Lieven Clement. 2020. "Robust       |
|               | Summarization and Inference in Proteome-Wide     |
|               | Label-Free Quantification." *Molecular &         |
|               | Cellular Proteomics: MCP* 19 (7): 1209--19.      |
|               |                                                  |
|               | [Bioconductor:                                   |
|               | msqrob2](https://www%20.bioc%20ond%20uc          |
|               | tor.org/packages/release/bioc/html/msqrob2.html) |
+---------------+--------------------------------------------------+

### Expression Analysis

#### Input Parameters

+--------------+--------------+--------------+--------------+
| parameter    | description  | inputs       | default      |
+==============+==============+==============+==============+
| log2fc_min   | the minimum  | *0 to Inf*   | *2*          |
|              | cutoff for   |              |              |
|              | displaying   |              |              |
+--------------+--------------+--------------+--------------+
| signifi      | the column   | -   a        | *a           |
| cance_column | to use for   |              | dj_p_value*  |
|              | significance | dj_p_value,  |              |
|              |              |              |              |
|              |              | p_value\*    |              |
+--------------+--------------+--------------+--------------+
| sign         | the minimum  | *1 to 1/Inf* | *0.05*       |
| ificance_max | cutoff for   |              |              |
|              | displaying   |              |              |
+--------------+--------------+--------------+--------------+
| l            | how to label | any value in | *protein*    |
| abels_column | the points   | annotations, |              |
|              |              | eg           |              |
|              |              | *gene_name,  |              |
|              |              | description* |              |
+--------------+--------------+--------------+--------------+
| show_pannels | show the     | *TRUE /      | *TRUE*       |
|              | colored      | FALSE*       |              |
|              | areas        |              |              |
|              | highlighting |              |              |
|              | significance |              |              |
+--------------+--------------+--------------+--------------+
| show_lines   | show the     | *TRUE /      | *TRUE*       |
|              | dotted       | FALSE*       |              |
|              | cutoff lines |              |              |
+--------------+--------------+--------------+--------------+
| s            | show the     | *TRUE /      | *TRUE*       |
| how_fc_scale | second scale | FALSE*       |              |
|              | on top       |              |              |
|              | converting   |              |              |
|              | *l           |              |              |
|              | og2          |              |              |
|              | _foldchange* |              |              |
|              | to           |              |              |
|              | *foldchange* |              |              |
+--------------+--------------+--------------+--------------+
| point_size   | size of each | *1/10* to    | *NULL*       |
|              | dot          | *5*          |              |
+--------------+--------------+--------------+--------------+
| co           | color for    | [any R       | *dodgerblue* |
| lor_positive | positive     | accepts]     |              |
|              | expression   | (htt%20ps:// |              |
|              |              | r-charts%20. |              |
|              |              | com/colors/) |              |
+--------------+--------------+--------------+--------------+
| co           | color for    | [any R       | *firebrick1* |
| lor_negative | negative     | accepts]     |              |
|              | expression   | (htt%20ps:// |              |
|              |              | r-charts%20. |              |
|              |              | com/colors/) |              |
+--------------+--------------+--------------+--------------+

```{r, message=FALSE}
library("dplyr")
library("tidyproteomics")

rdata <- hela_proteins %>% 
  normalize(.method = 'loess') %>%
  expression(knockdown/control)
```

#### Exporting Results

The results of this analysis are stored in the tidyproteomics
data-object and can easily be exported to save as a flat data table or
used in an external down stream analysis.

```{r}
rdata %>% 
  export_analysis(knockdown/control, .analysis = 'expression')
```

#### Volcano Plot

The default is to plot accoring to adjusted p-value with a Log2
fold-change cutoff of 2 or greater.

```{r}
rdata %>% 
  plot_volcano(knockdown/control)
```

The parameters can be adjusted to suit your needs ...

```{r}
rdata %>% 
  plot_volcano(knockdown/control,
               significance_column = 'p_value',
               significance_max = 0.01,
               log2fc_min = 2)
```

... or desires.

```{r}
library(ggplot2)

rdata %>% 
  plot_volcano(knockdown/control,
               significance_column = 'p_value',
               significance_max = 0.01,
               log2fc_min = 2,
               color_positive = 'orange',
               color_negative = 'purple',
               show_lines = FALSE,
               show_fc_scale = FALSE,
               show_pannels = FALSE,
               labels_column = 'gene_name') +
  labs(title = "A nice volcano plot", subtitle = "with great colors")
```

#### Proportional Plot

The proportional plot is complementary to the volcano plot and utilizes
all the same underlying data. The default is to show both the top 1%
(`proportion_min = 0.01`) along with the values passing the fold-change
and significance cutoffs.

```{r, message=FALSE}
library(ggplot2)

rdata %>% 
  plot_proportion(knockdown/control)
```

This plot can likewise also be manipulated like the volcano plot.

```{r}
library(ggplot2)

rdata %>% 
  plot_proportion(knockdown/control,
                  significance_column = 'p_value',
                  proportion_min = 0.1,
                  log2fc_min = 2,
                  color_positive = 'orange',
                  color_negative = 'purple',
                  show_lines = TRUE,
                  labels_column = 'gene_name')

```
