---
title: "Importing"
output: rmarkdown::html_vignette
description: |
  tidyproteomics: simplified data object and analysis functions
vignette: >
  %\VignetteIndexEntry{Importing}
  %\VignetteEngine{knitr::rmarkdown}
  %\VignetteEncoding{UTF-8}
---

```{r, echo = FALSE}
knitr::opts_chunk$set(collapse = TRUE, comment = "#>")
set.seed(1041)
options(dplyr.print_max = 10)
```

Importing data into tidyproteomics with `import()` currently accommodates data generated by [ProteomeDiscoverer](https://www.thermofisher.com/us/en/home/industrial/mass-spectrometry/liquid-chromatography-mass-spectrometry-lc-ms/lc-ms-software/multi-omics-data-analysis/proteome-discoverer-software.html), [MaxQuant](https://www.maxquant.org/), [Skyline](https://skyline.ms/project/home/software/skyline/begin.view) and [DIA-NN](https://github.com/vdemichev/DiaNN). Support for additional quantitative proteomics software suites can be added. The data needs to be quantitative, have descriptors for proteins and contain sample sources. Annotation are welcome at the import stage, but can be added later.

Table formatted data (eg. .csv, .tsv, .xlsx) from both ProteomeDiscoverer and MaxQuant meet the requirements, and are defined in the package data `tidyproteomics/inst/extdata/config/` accordingly. Note that the groups *sample*, *identifier* and *quantitative* are required, while the rest are optional, and only used if a match is found - currently this does not have good error handeling, so be aware. For ProteomeDiscoverer the peptide's config can be modified to use `Master Protein Accessions` or `Protein Accessions` if either column is present. Also note, that for some configurations the *sample* group has no defined "supplied name", as this is later derived by the extraction code.

```{r, message = FALSE}
library("dplyr")
library("tidyproteomics")
```

### ProteomeDiscoverer

The ProteomeDiscoverer software suite has the ability to export data post analysis for both `peptides` and `proteins`. This can be accomplished by opening the Study Results, then selecting `File \> Export \> To Microsoft Excel`. In the pop up keep only Level 1 checked, and select either "Proteins" or "Peptide Groups". See the Exporting section below to make sure the required columns are present.

#### Initial Set Up

When setting up your experiment it is essential to create sample names in the `Study Definition` tab of the open Study, navigate to the `Study Factors` box and simply add `Catigorical Factor` and name it (eg. My_Sample) and name your study groups (eg. WT and KO). These destinations need to be applied to the import files under the `Samples` tab. Notice the left most column is the added `Catigorical Factor` named the same as what was supplied. In each line there is a pull-down menu for designating each file (or label for TMT) to one of the `Catigorical Factors`. This will now ensure that when importing the data there are properly labeled sample. No big deal if that hasn't been done, it can be fixed with the `modify()` function.

#### Exporting

##### Proteins

The columns following columns should be considered. These can be modified in the file `tidyproteomics/inst/extdata/config/proteomediscoverer_protein_headers`:

| group        | normalized name     | supplied name      | required |
|--------------|---------------------|--------------------|----------|
| sample       | sample              |                    | yes      |
| sample       | sample_file         |                    | yes      |
| identifier   | protein_accession   | Accession          | yes      |
| quantitative | abundance           | Abundance:         | yes      |
| impute       | match_between_runs  | Found in Sample:   |          |
| accounting   | num_peptides        | \# Peptides        |          |
| accounting   | num_unique_peptides | \# Unique Peptides |          |
| accounting   | num_psms            | \# PSMs            |          |
| annotation   | description         | Description        |          |
| annotation   | gene_id_entrez      | Entrez Gene ID     |          |
| annotation   | gene_id_ensemble    | Ensembl Gene ID    |          |
| annotation   | gene_name           | Gene Symbol        |          |
| annotation   | chromosome          | Chromosome         |          |
| annotation   | biological_process  | Biological Process |          |
| annotation   | cellular_component  | Cellular Component |          |
| annotation   | molecular_function  | Molecular Function |          |
| annotation   | kegg_pathway        | KEGG Pathways      |          |
| annotation   | wiki_pathway        | WikiPathways       |          |
| annotation   | reactome_pathway    | Reactome Pathways  |          |

##### Peptides

The columns following columns should be considered. These can be modified in the file `tidyproteomics/inst/extdata/config/proteomediscoverer_peptide_headers`:

| group        | normalized name    | supplied name             | required |
|--------------|--------------------|---------------------------|----------|
| sample       | sample             |                           | yes      |
| sample       | sample_file        |                           | yes      |
| identifier   | protein_accession  | Master Protein Accessions | yes      |
| identifier   | peptide            | Annotated Sequence        | yes      |
| identifier   | modifications      | Modifications             | yes      |
| quantitative | abundance          | Abundance:                | yes      |
| impute       | match_between_runs | Found in Sample:          |          |

```{r, eval = FALSE}
# replace path_to_package_data("proteins") with the path to your local data.
# ecoli_proteins <- "./data/ecoli_export_table.xlsx" %>%
#    import("ProteomeDiscoverer", "proteins") 
ecoli_proteins <- path_to_package_data("ecoli-hint_proteins") %>%
   import("ProteomeDiscoverer", "proteins") 
```

### MaxQuant

The MaxQuant software suite creates files in project sub directories following `current_project \> combined \> txt` with data both *peptides (evidence.txt)* and *proteins (proteinGroups.txt)*.

#### Initial Set Up

While *evidence.txt* contains values for each imported file (important for comparative statistics), the *proteinGroups.txt* file will only contain an entry for each file if in the initial MaxQuant configuration the Experiment column in the raw data tab has a unique value for each file (eg 1, 2, 3, ...), otherwise the values get merged on common experiment groups in the output for the protein level data. The sample groups can then be set with the `modify()` function.

#### Exporting

##### Proteins

The columns following columns should be considered. These can be modified in the file `tidyproteomics/inst/extdata/config/MaxQuant_proteins.tsv`:

| category     | column_defined      | column_import           | required |
|--------------|---------------------|-------------------------|----------|
| sample       | sample              |                         | yes      |
| sample       | sample_file         |                         | yes      |
| identifier   | protein_accession   | Protein IDs             | yes      |
| quantitative | abundance           | Intensity\\s            | yes      |
| impute       | match_between_runs  |                         |          |
| accounting   | num_psms            | MS/MS count             |          |
| accounting   | num_peptides        | Peptide counts (all)    |          |
| accounting   | num_unique_peptides | Peptide counts (unique) |          |

##### Peptides

The columns following columns should be considered. These can be modified in the file `tidyproteomics/inst/extdata/config/MaxQuant_peptides.tsv`:

| category     | column_defined     | column_import | required |
|--------------|--------------------|---------------|----------|
| sample       | sample_file        | Raw file      | yes      |
| sample       | sample             | Experiment    | yes      |
| identifier   | protein_accession  | Proteins      | yes      |
| identifier   | peptide            | Sequence      | yes      |
| identifier   | modifications      | Modifications | yes      |
| quantitative | abundance          | Intensity\\s  | yes      |
| impute       | match_between_runs | Type          |          |
| accounting   | num_psms           | MS/MS count   |          |

```{r, eval = FALSE}
ecoli_proteins <- "path_to_maxquant_project/combined/txt/proteinGroups.txt" %>%
   import("MaxQuant", "proteins") %>%
   reassign(field = 'sample', pattern = 'sample_1', replace = 'hint_ko') %>%
   reassign(field = 'sample', pattern = 'sample_2', replace = 'hint_ko') %>%
   reassign(field = 'sample', pattern = 'sample_3', replace = 'hint_ko') %>%
   reassign(field = 'sample', pattern = 'sample_4', replace = 'hint_ko') %>%
   reassign(field = 'sample', pattern = 'sample_5', replace = 'wt') %>%
   reassign(field = 'sample', pattern = 'sample_6', replace = 'wt') %>%
   reassign(field = 'sample', pattern = 'sample_7', replace = 'wt') %>%
   reassign(field = 'sample', pattern = 'sample_8', replace = 'wt')
```

### Skyline

The Skyline software suite can export quantitative peptide data for most analyses.

#### Initial Set Up

A report need to be established, under `File > Export > Report`. Select `Edit list...`, `Group: > External Tools` then click `Add...`. Select the Columns that correspond to the required values shown below, name the report in `Report Name:` and click `OK`.

#### Exporting

##### Proteins

Not yet supported. Peptides can be combined into proteins with `rollup()`.

##### Peptides

The columns following columns should be considered. These can be modified in the file `tidyproteomics/inst/extdata/config/SkyLine_peptides.tsv`:

| category     | column_defined     | column_import     | required |
|--------------|--------------------|-------------------|----------|
| sample       | sample_file        | File Name         | yes      |
| sample       | sample             | Replicate Name    | yes      |
| identifier   | protein_accession  | Protein Accession | yes      |
| identifier   | peptide            | Peptide Sequence  | yes      |
| identifier   | modifications      | Modified Sequence | yes      |
| quantitative | abundance          | Normalized Area   | yes      |
| impute       | match_between_runs | Identified        | yes      |
| accounting   | num_psms           |                   | yes      |
| accounting   | is_decoy           | Is Decoy          | yes      |

```{r, eval = FALSE}
ecoli_proteins <- "path_to_skyline_project/output_file_name.csv" %>%
   import("Skyline", "peptides")
```

### DIA-NN

The DIA-NN software suite exports quantitative peptide data back into the project folder as *report.tsv*.

#### Exporting

##### Proteins

Not yet supported. Peptides can be combined into proteins with `rollup()`.

##### Peptides

The columns following columns should be considered. These can be modified in the file `tidyproteomics/inst/extdata/config/DIA-NN_peptides.tsv`:

| category     | column_defined     | column_import     | required |
|--------------|--------------------|-------------------|----------|
| sample       | sample_file        | File.Name         | yes      |
| sample       | sample             | Run               | yes      |
| identifier   | protein_accession  | Protein.Ids       | yes      |
| identifier   | peptide            | Stripped.Sequence | yes      |
| identifier   | modifications      | Modified.Sequence | yes      |
| quantitative | abundance          | Ms1.Area          | yes      |
| impute       | match_between_runs |                   | yes      |
| accounting   | num_psms           |                   | yes      |

```{r, eval = FALSE}
ecoli_proteins <- "path_to_diann_project/output_file_name.csv" %>%
   import("DIA-NN", "peptides")
```
