<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="
    Creates a simple, universal S3 data structure for the post 
    analysis of mass spectrometry based quantitative proteomic data. In
    addition, this package collects, adapts and organizes several useful
    algorithms and methods used in typical post analysis workflows.">
<title>An S3 data object and framework for common quantitative proteomic analyses • tidyproteomics</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="apple-touch-icon-60x60.png">
<script src="deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="pkgdown.js"></script><meta property="og:title" content="An S3 data object and framework for common quantitative proteomic analyses">
<meta property="og:description" content="
    Creates a simple, universal S3 data structure for the post 
    analysis of mass spectrometry based quantitative proteomic data. In
    addition, this package collects, adapts and organizes several useful
    algorithms and methods used in typical post analysis workflows.">
<meta property="og:image" content="https://jeffsocal.github.io/tidyproteomics/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="index.html">tidyproteomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.8.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="articles/overview.html">Overview</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="articles/updates.html">Updates</a>
    <a class="dropdown-item" href="articles/data-object.html">The Data-Object</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Basics</h6>
    <a class="dropdown-item" href="articles/importing.html">Importing</a>
    <a class="dropdown-item" href="articles/annotating.html">Annotating</a>
    <a class="dropdown-item" href="articles/summarizing.html">Summarizing</a>
    <a class="dropdown-item" href="articles/visualizing.html">Visualizing</a>
    <a class="dropdown-item" href="articles/extracting.html">Extracting</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Processing</h6>
    <a class="dropdown-item" href="articles/subsetting.html">Subsetting</a>
    <a class="dropdown-item" href="articles/normalizing.html">Normalizing</a>
    <a class="dropdown-item" href="articles/imputing.html">Imputing</a>
    <a class="dropdown-item" href="articles/automating.html">Automating</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Analysis</h6>
    <a class="dropdown-item" href="articles/expression.html">Expression</a>
    <a class="dropdown-item" href="articles/enrichment.html">Enrichment</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Peptides to Proteins</h6>
    <a class="dropdown-item" href="articles/collapsing.html">Collapsing</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Example Workflows</h6>
    <a class="dropdown-item" href="articles/workflow-simple.html">Simple</a>
    <a class="dropdown-item" href="articles/workflow-importing.html">Importing Demo</a>
    <a class="dropdown-item" href="articles/workflow-publication.html">Publication Demo</a>
    <a class="dropdown-item" href="articles/workflow-userimport.html">User Defined Import</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="articles/updates.html">Updates</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-home">
<div class="row">
  <main id="main" class="col-md-9"><div class="section level1">
<div class="page-header">
<img src="logo.png" class="logo" alt=""><h1 id="tidyproteomics-">tidyproteomics <a class="anchor" aria-label="anchor" href="#tidyproteomics-"></a>
</h1>
</div>
<p>An R package for the tidy-ing, post processing and analysis of quantitative proteomic data.</p>
<p>Proteomics analysis software, available either through a paid subscription or as an open-source tool, fail to output data in a well conceived tidy format. A majority of these tools generate output formats that have either mixed wide- and long-format data structures, columns headers with messy names and added symbols, and often confusing variable names. This leads researchers to create one-off scripts for cleaning and importing data from various formats, often creating an environment of unmaintained, bespoke code. This package attempts to solve that problem by creating a flexible import tool to unify multiple formats and create an new tidy R object for proteomics analysis.</p>
<p>This package supports at a high level:</p>
<ul>
<li>data importing</li>
<li>data filtering</li>
<li>data visualization</li>
<li>quantitative normalization &amp; imputation</li>
<li>two-sample expression &amp; term enrichment analysis</li>
<li>protein inference, sequence coverage and visualization</li>
</ul>
<div class="section level2">
<h2 id="links">Links<a class="anchor" aria-label="anchor" href="#links"></a>
</h2>
<div class="section level3">
<h3 id="publication">Publication<a class="anchor" aria-label="anchor" href="#publication"></a>
</h3>
<p>Jones, J., MacKrell, E.J., Wang, TY. et al. <em>Tidyproteomics: an open-source R package and data object for quantitative proteomics post analysis and visualization.</em> <a href="https://bmcbioinformatics.biomedcentral.com/articles/10.1186/s12859-023-05360-7" class="external-link">BMC Bioinformatics 24, 239 (2023)</a></p>
</div>
<div class="section level3">
<h3 id="r-package">R Package<a class="anchor" aria-label="anchor" href="#r-package"></a>
</h3>
<ul>
<li>Code: <a href="https://github.com/jeffsocal/tidyproteomics" class="external-link">github.com/jeffsocal/tidyproteomics</a>
</li>
<li>Docs: <a href="https://jeffsocal.github.io/tidyproteomics/">jeffsocal.github.io/tidyproteomics/</a>
</li>
</ul>
</div>
<div class="section level3">
<h3 id="rshiny-app">RShiny App<a class="anchor" aria-label="anchor" href="#rshiny-app"></a>
</h3>
<ul>
<li>Code: <a href="https://github.com/ejmackrell/tidyproteomics-interactive" class="external-link">github.com/ejmackrell/tidyproteomics-interactive</a>
</li>
<li>Web : <a href="http://bioinformatics.pel.caltech.edu/tidyproteomics/" class="external-link">bioinformatics.pel.caltech.edu/tidyproteomics/</a>
</li>
<li>Docs: <a href="https://ejmackrell.github.io/tidyproteomics-interactive/" class="external-link">ejmackrell.github.io/tidyproteomics-interactive/</a>
</li>
</ul>
</div>
</div>
<div class="section level2">
<h2 id="data-support">Data Support<a class="anchor" aria-label="anchor" href="#data-support"></a>
</h2>
<p>Importing is currently implemented for a few platforms and assume peptide level FDR (at the user’s desired level) has already been accounted for. See <code><a href="articles/importing.html">vignette("importing")</a></code>. Importing is flexible enough to accept other data platforms in flat files (.csv, .tsv, and .xlsx) with a custom configuration.</p>
<table class="table">
<colgroup>
<col width="24%">
<col width="26%">
<col width="24%">
<col width="24%">
</colgroup>
<thead><tr class="header">
<th>Platform</th>
<th>peptides</th>
<th>proteins</th>
<th>notes</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>ProteomeDiscoverer</td>
<td>*.xlsx <em>peptides export</em>
</td>
<td>*.xlsx <em>proteins export</em>
</td>
<td>requires layout configuration</td>
</tr>
<tr class="even">
<td>MaxQuant</td>
<td>evidence.txt</td>
<td>proteinGroups.txt</td>
<td></td>
</tr>
<tr class="odd">
<td>FragPipe</td>
<td>combined_peptide.tsv</td>
<td>combined_protein.tsv</td>
<td></td>
</tr>
<tr class="even">
<td>Skyline</td>
<td>*.csv <em>MSstats peptide report</em>
</td>
<td></td>
<td>requires MSstats install</td>
</tr>
<tr class="odd">
<td>DIA-NN</td>
<td>*.tsv <em>peptide report</em>
</td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>mzTab</td>
<td>*.mzTab (v1.0.0)</td>
<td>*.mzTab (v1.0.0)</td>
<td>does not track MBR</td>
</tr>
</tbody>
</table>
</div>
<div class="section level2">
<h2 id="ease-of-use">Ease of Use<a class="anchor" aria-label="anchor" href="#ease-of-use"></a>
</h2>
<p>This package supports the same <a href="https://en.wikipedia.org/wiki/Syntactic_sugar" class="external-link">syntactic sugar</a> utilized in the tidy-verse functions like filter, and introduces the <code>%like%</code> operator, see <code><a href="articles/subsetting.html">vignette("subsetting")</a></code> . These operations can extend to all aspects of the data set, including sample names, protein IDs, annotations and accountings like <em>match_between_runs</em> and <em>num_peptides</em>.</p>
<table class="table">
<colgroup>
<col width="26%">
<col width="26%">
<col width="47%">
</colgroup>
<thead><tr class="header">
<th>operator</th>
<th>description</th>
<th>example</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>==</td>
<td>equals</td>
<td>
<code>sample == 'wt'</code> , <code>match_between_runs == FALSE</code>
</td>
</tr>
<tr class="even">
<td>!=</td>
<td>does not equal</td>
<td><code>biological_function != 'DNA metabolism'</code></td>
</tr>
<tr class="odd">
<td>&lt;, &gt;</td>
<td>less, greater than</td>
<td><code>num_unique_peptides &gt;= 2</code></td>
</tr>
<tr class="even">
<td>%like%</td>
<td>contains</td>
<td><code>description %like% 'ribosome'</code></td>
</tr>
<tr class="odd">
<td>! %like%</td>
<td>does not contain</td>
<td><code>!description %like% 'ribosome'</code></td>
</tr>
<tr class="even">
<td>/</td>
<td>ratio <em>(expression)</em>
</td>
<td><code>experiment / control</code></td>
</tr>
</tbody>
</table>
<p>Expression analysis also utilizes this type of syntax when referencing samples for analysis. For example <code>data %&gt;% expression(knockdown/control)</code> would know to run the differential expression of the sample <em>ko</em> with respect to the sample <em>wt</em> such that positive log2 difference would be up-expressed in <em>ko</em> and a negative log2 differences would be down-expressed in <em>ko</em>.</p>
</div>
<div class="section level2">
<h2 id="nomenclature">Nomenclature<a class="anchor" aria-label="anchor" href="#nomenclature"></a>
</h2>
<p>The tidyprotoemics package uses a conserved naming convention that facilitates a “contract” with funtions that operate on the data, making it easier to maintain and modify the code. This naming convention ensures that functions are generic to their inputs, being able to operate on a variety of data without having to explicitly name each column and variable which may be named different between data sets. All of the effort of standardizing the variable names in the tidyproteomics data object is done upfront when importing the data. See <code><a href="articles/importing.html">vignette("importing")</a></code>.</p>
<p>This package divides data into tables similar to those in SQL, making it easy to organize and analyze data. This table structure offers a convenient way to explore relationships between different variables and to filter data according to specific criteria.</p>
<div class="section level3">
<h3 id="experimental-table">Experimental Table<a class="anchor" aria-label="anchor" href="#experimental-table"></a>
</h3>
<table class="table">
<colgroup>
<col width="25%">
<col width="75%">
</colgroup>
<thead><tr class="header">
<th>Variable</th>
<th>Reference</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>sample_id</td>
<td>A checksum id on the individual data imported, allows for an unambiguous identifier usefull in differentiating samples of the same name</td>
</tr>
<tr class="even">
<td>import_file</td>
<td>The file referenced at import, allows for multiple files to be merged</td>
</tr>
<tr class="odd">
<td>sample_file</td>
<td>The individual MS file, or for TMT the channel, that contains data for a single observation of a given protein</td>
</tr>
<tr class="even">
<td>sample</td>
<td>The name of the MS data collected for a given class, collection, of a specimen or a cell culture, etc.</td>
</tr>
<tr class="odd">
<td>replicate</td>
<td>A tidyproteomics derived integer assignment to a collection of same-named samples</td>
</tr>
</tbody>
</table>
<p><u>Example</u></p>
<div class="sourceCode" id="cb1"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb1-1"><a href="#cb1-1" tabindex="-1"></a><span class="fu">library</span>(tidyproteomics)</span>
<span id="cb1-2"><a href="#cb1-2" tabindex="-1"></a></span>
<span id="cb1-3"><a href="#cb1-3" tabindex="-1"></a>hela_proteins<span class="sc">$</span>experiments</span>
<span id="cb1-4"><a href="#cb1-4" tabindex="-1"></a></span>
<span id="cb1-5"><a href="#cb1-5" tabindex="-1"></a><span class="co"># A tibble: 6 × 5</span></span>
<span id="cb1-6"><a href="#cb1-6" tabindex="-1"></a>  sample_id import_file                sample_file sample    replicate</span>
<span id="cb1-7"><a href="#cb1-7" tabindex="-1"></a>  <span class="sc">&lt;</span>chr<span class="sc">&gt;</span>     <span class="er">&lt;</span>chr<span class="sc">&gt;</span>                      <span class="er">&lt;</span>chr<span class="sc">&gt;</span>       <span class="er">&lt;</span>chr<span class="sc">&gt;</span>     <span class="er">&lt;</span>chr<span class="sc">&gt;</span>    </span>
<span id="cb1-8"><a href="#cb1-8" tabindex="-1"></a><span class="dv">1</span> e9b20ea7  p97KD_HCT116_proteins.xlsx f1          control   <span class="dv">1</span>        </span>
<span id="cb1-9"><a href="#cb1-9" tabindex="-1"></a><span class="dv">2</span> ef79cc4c  p97KD_HCT116_proteins.xlsx f4          control   <span class="dv">2</span>        </span>
<span id="cb1-10"><a href="#cb1-10" tabindex="-1"></a><span class="dv">3</span> eebba67b  p97KD_HCT116_proteins.xlsx f5          control   <span class="dv">3</span>        </span>
<span id="cb1-11"><a href="#cb1-11" tabindex="-1"></a><span class="dv">4</span> ebf4b0fe  p97KD_HCT116_proteins.xlsx f2          knockdown <span class="dv">1</span>        </span>
<span id="cb1-12"><a href="#cb1-12" tabindex="-1"></a><span class="dv">5</span> ea36dac9  p97KD_HCT116_proteins.xlsx f3          knockdown <span class="dv">2</span>        </span>
<span id="cb1-13"><a href="#cb1-13" tabindex="-1"></a><span class="dv">6</span> ecfd1822  p97KD_HCT116_proteins.xlsx f6          knockdown <span class="dv">3</span>   </span></code></pre></div>
</div>
<div class="section level3">
<h3 id="quantitation-table">Quantitation Table<a class="anchor" aria-label="anchor" href="#quantitation-table"></a>
</h3>
<table class="table">
<colgroup>
<col width="25%">
<col width="75%">
</colgroup>
<thead><tr class="header">
<th>Variable</th>
<th>Reference</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>sample_id</td>
<td>A checksum id on the individual data imported, allows for an unambiguous identifier useful in differentiating samples of the same name</td>
</tr>
<tr class="even">
<td>sample</td>
<td>The name of the MS data collected for a given class, collection of sampling of tissue, a culture etc.</td>
</tr>
<tr class="odd">
<td>replicate</td>
<td>A tidyproteomics derived integer assignment to a collection of same-named samples</td>
</tr>
<tr class="even">
<td><em>identifier</em></td>
<td>The identifier of the “thing” being measured.</td>
</tr>
<tr class="odd">
<td>protein</td>
<td>In the case of protein-level data, this is simply <code>protein</code> and is usually populated with UniProt accession numbers, names, or other protein identifiers.</td>
</tr>
<tr class="even">
<td>protein, peptide, modification</td>
<td>In the case of peptide-level data, this is a multiple of <code>protein</code> (accession number), <code>peptide</code> (amino acid sequence) and <code>modification</code> (post-translational modification).</td>
</tr>
<tr class="odd">
<td>abundance_raw</td>
<td>The imported quantitative measure for each protein or peptide. The tidyproteomics data object stores abundance values for the raw data and each normalization. This allows for the direct comparison of normalization methods against the raw data, and for one set of abundance values to be used for subsequent analyses.</td>
</tr>
</tbody>
</table>
<p><u>Example</u></p>
<div class="sourceCode" id="cb2"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb2-1"><a href="#cb2-1" tabindex="-1"></a><span class="fu">library</span>(tidyproteomics)</span>
<span id="cb2-2"><a href="#cb2-2" tabindex="-1"></a></span>
<span id="cb2-3"><a href="#cb2-3" tabindex="-1"></a>hela_proteins<span class="sc">$</span>quantitation</span>
<span id="cb2-4"><a href="#cb2-4" tabindex="-1"></a></span>
<span id="cb2-5"><a href="#cb2-5" tabindex="-1"></a><span class="co"># A tibble: 42,330 × 5</span></span>
<span id="cb2-6"><a href="#cb2-6" tabindex="-1"></a>   sample_id sample    replicate protein abundance_raw</span>
<span id="cb2-7"><a href="#cb2-7" tabindex="-1"></a>   <span class="sc">&lt;</span>chr<span class="sc">&gt;</span>     <span class="er">&lt;</span>chr<span class="sc">&gt;</span>     <span class="er">&lt;</span>chr<span class="sc">&gt;</span>     <span class="er">&lt;</span>chr<span class="sc">&gt;</span>           <span class="er">&lt;</span>dbl<span class="sc">&gt;</span></span>
<span id="cb2-8"><a href="#cb2-8" tabindex="-1"></a> <span class="dv">1</span> e9b20ea7  control   <span class="dv">1</span>         Q15149    <span class="fl">1011259992.</span></span>
<span id="cb2-9"><a href="#cb2-9" tabindex="-1"></a> <span class="dv">2</span> ef79cc4c  control   <span class="dv">2</span>         Q15149    <span class="fl">1093277593.</span></span>
<span id="cb2-10"><a href="#cb2-10" tabindex="-1"></a> <span class="dv">3</span> eebba67b  control   <span class="dv">3</span>         Q15149     <span class="fl">980809516.</span></span>
<span id="cb2-11"><a href="#cb2-11" tabindex="-1"></a> <span class="dv">4</span> ebf4b0fe  knockdown <span class="dv">1</span>         Q15149    <span class="fl">1410445367.</span></span>
<span id="cb2-12"><a href="#cb2-12" tabindex="-1"></a> <span class="dv">5</span> ea36dac9  knockdown <span class="dv">2</span>         Q15149    <span class="fl">1072305561.</span></span>
<span id="cb2-13"><a href="#cb2-13" tabindex="-1"></a> <span class="dv">6</span> ecfd1822  knockdown <span class="dv">3</span>         Q15149    <span class="fl">1486561518.</span></span>
<span id="cb2-14"><a href="#cb2-14" tabindex="-1"></a><span class="co"># … with 42,324 more rows</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="annotation-table">Annotation Table<a class="anchor" aria-label="anchor" href="#annotation-table"></a>
</h3>
<table class="table">
<colgroup>
<col width="25%">
<col width="75%">
</colgroup>
<thead><tr class="header">
<th>Variable</th>
<th>Reference</th>
</tr></thead>
<tbody>
<tr class="odd">
<td><em>identifier</em></td>
<td>The identifier of the “thing” being measured.</td>
</tr>
<tr class="even">
<td>term</td>
<td>The name of the collection of annotations, such as gene_name or description.</td>
</tr>
<tr class="odd">
<td>annotation</td>
<td>The descriptive annotation.</td>
</tr>
</tbody>
</table>
<p><u>Example</u></p>
<div class="sourceCode" id="cb3"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb3-1"><a href="#cb3-1" tabindex="-1"></a><span class="fu">library</span>(tidyproteomics)</span>
<span id="cb3-2"><a href="#cb3-2" tabindex="-1"></a></span>
<span id="cb3-3"><a href="#cb3-3" tabindex="-1"></a><span class="fu">print</span>(hela_proteins<span class="sc">$</span>annotations)</span>
<span id="cb3-4"><a href="#cb3-4" tabindex="-1"></a></span>
<span id="cb3-5"><a href="#cb3-5" tabindex="-1"></a><span class="co"># A tibble: 63,495 × 3</span></span>
<span id="cb3-6"><a href="#cb3-6" tabindex="-1"></a>   protein term               annotation                                                                                                          </span>
<span id="cb3-7"><a href="#cb3-7" tabindex="-1"></a>   <span class="sc">&lt;</span>chr<span class="sc">&gt;</span>   <span class="er">&lt;</span>chr<span class="sc">&gt;</span>              <span class="er">&lt;</span>chr<span class="sc">&gt;</span>                                                                                                               </span>
<span id="cb3-8"><a href="#cb3-8" tabindex="-1"></a> <span class="dv">1</span> Q15149  description        Plectin OS<span class="ot">=</span>Homo sapiens OX<span class="ot">=</span><span class="dv">9606</span> GN<span class="ot">=</span>PLEC PE<span class="ot">=</span><span class="dv">1</span> SV<span class="ot">=</span><span class="dv">3</span>                                                                   </span>
<span id="cb3-9"><a href="#cb3-9" tabindex="-1"></a> <span class="dv">2</span> Q15149  gene_id_entrez     <span class="dv">5339</span>                                                                                                                </span>
<span id="cb3-10"><a href="#cb3-10" tabindex="-1"></a> <span class="dv">3</span> Q15149  gene_id_ensemble   ENSG00000178209                                                                                                     </span>
<span id="cb3-11"><a href="#cb3-11" tabindex="-1"></a> <span class="dv">4</span> Q15149  gene_name          PLEC                                                                                                                </span>
<span id="cb3-12"><a href="#cb3-12" tabindex="-1"></a> <span class="dv">5</span> Q15149  biological_process cell differentiation;cell growth;cellular homeostasis;coagulation;defense response;metabolic process</span>
<span id="cb3-13"><a href="#cb3-13" tabindex="-1"></a><span class="co"># … with 63,490 more rows</span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="section"></h3>
</div>
</div>
<div class="section level2">
<h2 id="installation">Installation<a class="anchor" aria-label="anchor" href="#installation"></a>
</h2>
<p>To install, open R and type:</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"devtools"</span><span class="op">)</span></span>
<span><span class="fu">devtools</span><span class="fu">::</span><span class="fu"><a href="https://remotes.r-lib.org/reference/install_github.html" class="external-link">install_github</a></span><span class="op">(</span><span class="st">"jeffsocal/tidyproteomics"</span><span class="op">)</span></span></code></pre></div>
<p>You will also need the Bioconductor packages <a href="https://bioconductor.org/packages/release/bioc/html/limma.html" class="external-link">limma</a>, <a href="https://bioconductor.org/packages/release/bioc/html/qvalue.html" class="external-link">qvalue</a>, and <a href="https://bioconductor.org/packages/release/bioc/html/fgsea.html" class="external-link">fgsea</a>, to install these type:</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="fu"><a href="https://rdrr.io/r/utils/install.packages.html" class="external-link">install.packages</a></span><span class="op">(</span><span class="st">"BiocManager"</span><span class="op">)</span></span>
<span><span class="fu">BiocManager</span><span class="fu">::</span><span class="fu"><a href="https://bioconductor.github.io/BiocManager/reference/install.html" class="external-link">install</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"limma"</span>,<span class="st">"qvalue"</span>,<span class="st">"fgsea"</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><strong>NOTE:</strong> There are several other packages required that will be prompted and automatically downloaded from CRAN when installing. Depending on your current system some packages require the installation of OS level libraries for advanced math and string manipulation.</p>
</div>
<div class="section level2">
<h2 id="get-started">Get Started<a class="anchor" aria-label="anchor" href="#get-started"></a>
</h2>
<p>If you want to test the features and capabilities of this project and get a better understanding of how it works, you can experiment with the two datasets that are included within the package. These datasets have been specially selected to provide you with the necessary working examples that will help you to get started with the project.</p>
<p>The first dataset is a protein-level dataset, and the second one is a small peptide-level dataset. Both datasets are easily accessible after loading the package. You can use these datasets to practice and explore the different features and functionalities of the package. They are also used throughout the vignettes and code documentation to provide you with clear and concise examples.</p>
<p>Protein level data.</p>
<div class="sourceCode" id="cb6"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb6-1"><a href="#cb6-1" tabindex="-1"></a><span class="fu">library</span>(tidyproteomics)</span>
<span id="cb6-2"><a href="#cb6-2" tabindex="-1"></a></span>
<span id="cb6-3"><a href="#cb6-3" tabindex="-1"></a>hela_protiens</span>
<span id="cb6-4"><a href="#cb6-4" tabindex="-1"></a></span>
<span id="cb6-5"><a href="#cb6-5" tabindex="-1"></a>── Quantitative Proteomics Data Object ──</span>
<span id="cb6-6"><a href="#cb6-6" tabindex="-1"></a></span>
<span id="cb6-7"><a href="#cb6-7" tabindex="-1"></a>Origin          ProteomeDiscoverer </span>
<span id="cb6-8"><a href="#cb6-8" tabindex="-1"></a>                <span class="fu">proteins</span> (<span class="fl">10.76</span> MB) </span>
<span id="cb6-9"><a href="#cb6-9" tabindex="-1"></a>Composition     <span class="dv">6</span> files </span>
<span id="cb6-10"><a href="#cb6-10" tabindex="-1"></a>                <span class="dv">2</span> <span class="fu">samples</span> (control, knockdown) </span>
<span id="cb6-11"><a href="#cb6-11" tabindex="-1"></a>Quantitation    <span class="dv">7055</span> proteins </span>
<span id="cb6-12"><a href="#cb6-12" tabindex="-1"></a>                <span class="dv">4</span> log10 dynamic range </span>
<span id="cb6-13"><a href="#cb6-13" tabindex="-1"></a>                <span class="fl">28.8</span>% missing values </span>
<span id="cb6-14"><a href="#cb6-14" tabindex="-1"></a><span class="fu">Accounting</span>      (<span class="dv">4</span>) match_between_runs num_peptides num_unique_peptides num_psms </span>
<span id="cb6-15"><a href="#cb6-15" tabindex="-1"></a><span class="fu">Annotations</span>     (<span class="dv">9</span>) description gene_id_entrez gene_id_ensemble gene_name biological_process</span>
<span id="cb6-16"><a href="#cb6-16" tabindex="-1"></a>                cellular_component molecular_function wiki_pathway</span>
<span id="cb6-17"><a href="#cb6-17" tabindex="-1"></a>                reactome_pathway </span>
<span id="cb6-18"><a href="#cb6-18" tabindex="-1"></a>                </span></code></pre></div>
<p>Peptide level data.</p>
<div class="sourceCode" id="cb7"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb7-1"><a href="#cb7-1" tabindex="-1"></a><span class="fu">library</span>(tidyproteomics)</span>
<span id="cb7-2"><a href="#cb7-2" tabindex="-1"></a></span>
<span id="cb7-3"><a href="#cb7-3" tabindex="-1"></a>hela_peptides</span>
<span id="cb7-4"><a href="#cb7-4" tabindex="-1"></a>── Quantitative Proteomics Data Object ──</span>
<span id="cb7-5"><a href="#cb7-5" tabindex="-1"></a></span>
<span id="cb7-6"><a href="#cb7-6" tabindex="-1"></a>Origin          ProteomeDiscoverer </span>
<span id="cb7-7"><a href="#cb7-7" tabindex="-1"></a>                <span class="fu">peptides</span> (<span class="fl">188.87</span> kB) </span>
<span id="cb7-8"><a href="#cb7-8" tabindex="-1"></a>Composition     <span class="dv">6</span> files </span>
<span id="cb7-9"><a href="#cb7-9" tabindex="-1"></a>                <span class="dv">2</span> <span class="fu">samples</span> (control, knockdown) </span>
<span id="cb7-10"><a href="#cb7-10" tabindex="-1"></a>Quantitation    <span class="dv">8</span> proteins </span>
<span id="cb7-11"><a href="#cb7-11" tabindex="-1"></a>                <span class="fl">3.3</span> log10 dynamic range </span>
<span id="cb7-12"><a href="#cb7-12" tabindex="-1"></a>                <span class="fl">21.3</span>% missing values </span>
<span id="cb7-13"><a href="#cb7-13" tabindex="-1"></a><span class="fu">Accounting</span>      (<span class="dv">3</span>) modifications match_between_runs num_psms </span></code></pre></div>
</div>
<div class="section level2">
<h2 id="import-your-data">Import Your Data<a class="anchor" aria-label="anchor" href="#import-your-data"></a>
</h2>
<p>Its simple to get started. Make a new project, drop your raw data in a folder labeled <em>data</em>. For more information see <code><a href="articles/workflow-simple.html">vignette("workflow-simple")</a></code></p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://jeffsocal.github.io/tidyproteomics/">tidyproteomics</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">data</span> <span class="op">&lt;-</span> <span class="st">"./data/some_ProteomeDiscoverer_data.xlsx"</span> <span class="op">%&gt;%</span></span>
<span>  <span class="fu"><a href="reference/import.html">import</a></span><span class="op">(</span><span class="st">"ProteomeDiscoverer"</span>, <span class="st">"proteins"</span><span class="op">)</span></span>
<span>  </span>
<span><span class="va">data</span> <span class="op">%&gt;%</span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="st">"samples"</span><span class="op">)</span></span></code></pre></div>
</div>
</div>
  </main><aside class="col-md-3"><div class="license">
<h2 data-toc-skip>License</h2>
<ul class="list-unstyled">
<li><a href="LICENSE.html">Full license</a></li>
<li><small><a href="https://opensource.org/licenses/mit-license.php" class="external-link">MIT</a> + file <a href="LICENSE-text.html">LICENSE</a></small></li>
</ul>
</div>


<div class="citation">
<h2 data-toc-skip>Citation</h2>
<ul class="list-unstyled">
<li><a href="authors.html#citation">Citing tidyproteomics</a></li>
</ul>
</div>

<div class="developers">
<h2 data-toc-skip>Developers</h2>
<ul class="list-unstyled">
<li>Jeff Jones <br><small class="roles"> Author, maintainer </small> <a href="https://orcid.org/0000-0002-7142-2222" target="orcid.widget" aria-label="ORCID" class="external-link"><span class="fab fa-orcid orcid" aria-hidden="true"></span></a> </li>
</ul>
</div>



  </aside>
</div>


    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jeff Jones.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
