<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="tidyproteomics: simplified data object and analysis functions
">
<title>Protein Accounting â€¢ tidyproteomics</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Protein Accounting">
<meta property="og:description" content="tidyproteomics: simplified data object and analysis functions
">
<meta property="og:image" content="https://jeffsocal.github.io/tidyproteomics/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">tidyproteomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.10</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/overview.html">Overview</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/updates.html">Updates</a>
    <a class="dropdown-item" href="../articles/data-object.html">The Data-Object</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Operations</h6>
    <a class="dropdown-item" href="../articles/importing.html">Importing</a>
    <a class="dropdown-item" href="../articles/summarizing.html">Summarizing</a>
    <a class="dropdown-item" href="../articles/subsetting.html">Subsetting</a>
    <a class="dropdown-item" href="../articles/normalizing.html">Normalizing</a>
    <a class="dropdown-item" href="../articles/imputing.html">Imputing</a>
    <a class="dropdown-item" href="../articles/annotating.html">Annotating</a>
    <a class="dropdown-item" href="../articles/collapsing.html">Protein Accounting</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Analysis</h6>
    <a class="dropdown-item" href="../articles/visualizing.html">Visualizing</a>
    <a class="dropdown-item" href="../articles/expression.html">Expression</a>
    <a class="dropdown-item" href="../articles/enrichment.html">Enrichment</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Workflows</h6>
    <a class="dropdown-item" href="../articles/workflow-simple.html">Simple</a>
    <a class="dropdown-item" href="../articles/workflow-importing.html">Importing Demo</a>
    <a class="dropdown-item" href="../articles/workflow-userimport.html">User Defined Import</a>
    <a class="dropdown-item" href="../articles/workflow-publication.html">Publication Demo</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Protein Accounting</h1>
            
      
      
      <div class="d-none name"><code>collapsing.Rmd</code></div>
    </div>

    
    
<p>Protein accounting is an important aspect of quantiative proteomics.
It consists of aggregating individual peptide measurements into a single
reported protein. However, sequence homology make this a less than
trivial process, as the LCMS process does not identify proteins
specifically, rather their individual peptide segments. It is then up to
the researcher to implement a method that infers the original protein
content.</p>
<p>While the proteomics community at large has devised several schemes
for protein accounting, not all methods are universally amenable. For
example, when accounting for cellular differences between to cell
states, it is often preferred to arrive at the minimal number of protein
sequences that can explain the data. That means tossing aside
identifiers to sequence variants - something researchers attempting to
identify specific genes might like to retain. Furthermore, for
researchers attempting to grasp if gene insertions are viable, gross
summation of peptide abundances may miss nuances in sequence variations.
These latter points are not well controlled for in platforms that cater
for proteome wide discoveries. Therefor, we have attempted to provide
several simple mechanisms that can apply to a variety of LCMS proteomics
projects both larger proteome wide discoveries and smaller induced
expression projects.</p>
<p>More sophisticated methods of protein accounting should utilize those
current implementations, export the quantitative protein level data and
then import into tidyproteomics.</p>
<p>NOTE: tidyproteomics allows you to group and summarize quantitative
values for any annotation term, such as gene_name or even
molecular_function giving you flexibility on how individual peptide
values get summarized into larger observations. See <a href="#collapsing-to-gene-name">Collapsing to Gene Name</a>.</p>
<div class="section level2">
<h2 id="the-inputs">The inputs<a class="anchor" aria-label="anchor" href="#the-inputs"></a>
</h2>
<div class="section level4">
<h4 id="assign_by">assign_by<a class="anchor" aria-label="anchor" href="#assign_by"></a>
</h4>
<p>The <code><a href="../reference/collapse.html">collapse()</a></code> function accommodates four methods for
accounting with the variable:</p>
<table class="table">
<colgroup>
<col width="26%">
<col width="73%">
</colgroup>
<thead><tr class="header">
<th>Method</th>
<th>Description</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>all-possible</td>
<td>retain all protein assignments regardless of homology</td>
</tr>
<tr class="even">
<td>non-homologous</td>
<td>retain all protein assignments, but only use quantitative values
from peptides that identify a single unique protein.</td>
</tr>
<tr class="odd">
<td>razor-local</td>
<td>peptide will be assigned to the protein group with the largest
number of total peptides identified within a sample group</td>
</tr>
<tr class="even">
<td>razor-global</td>
<td>peptide will be assigned to the protein group with the largest
number of total peptides identified globally</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="top_n">top_n<a class="anchor" aria-label="anchor" href="#top_n"></a>
</h4>
<p>this variable sets the number of peptide quantitative values to
include in the final protein quantitative value.</p>
</div>
<div class="section level4">
<h4 id="split_abundance">split_abundance<a class="anchor" aria-label="anchor" href="#split_abundance"></a>
</h4>
<p>a true/false variable, if true, will split abundances of
razor-peptides by the proportional abundance between proteins. This
method is experimental, has not been validated or peer-reviewed.</p>
<p>fasta_path can be provided to calculate iBAQ based values. However,
currently this method only supports Tryptic based peptide
accounting.</p>
</div>
</div>
<div class="section level2">
<h2 id="examples">Examples<a class="anchor" aria-label="anchor" href="#examples"></a>
</h2>
<p>Here we import a data-object of ProteomeDiscovererâ€™s protein level
accounting and investigate the difference between ProteomeDiscovererâ€™s
protein accounting and that implemented by <em>tidyproteomics</em>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="co"># download the data</span></span>
<span><span class="va">url_pro</span> <span class="op">&lt;-</span> <span class="st">"https://data.caltech.edu/records/aevwq-2ps50/files/ProteomeDiscoverer_2.5_p97KD_HCT116_proteins.xlsx?download=1"</span></span>
<span><span class="va">url_pep</span> <span class="op">&lt;-</span> <span class="st">"https://data.caltech.edu/records/aevwq-2ps50/files/ProteomeDiscoverer_2.5_p97KD_HCT116_peptides.xlsx?download=1"</span></span>
<span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">url_pro</span>, destfile <span class="op">=</span> <span class="st">"./data/pd_proteins.xlsx"</span><span class="op">)</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">url_pep</span>, destfile <span class="op">=</span> <span class="st">"./data/pd_peptides.xlsx"</span><span class="op">)</span></span>
<span></span>
<span><span class="co"># import the data</span></span>
<span><span class="va">pro_data</span> <span class="op">&lt;-</span> <span class="st">"./data/pd_proteins.xlsx"</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/import.html">import</a></span><span class="op">(</span><span class="st">'ProteomeDiscoverer'</span>, <span class="st">'proteins'</span><span class="op">)</span></span>
<span><span class="va">pep_data</span> <span class="op">&lt;-</span> <span class="st">"./data/pd_peptides.xlsx"</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/import.html">import</a></span><span class="op">(</span><span class="st">'ProteomeDiscoverer'</span>, <span class="st">'peptides'</span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="assign-peptides-by-razor-logic">Assign peptides by razor logic<a class="anchor" aria-label="anchor" href="#assign-peptides-by-razor-logic"></a>
</h3>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pro_frompep_razor</span> <span class="op">&lt;-</span> <span class="va">pep_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/collapse.html">collapse</a></span><span class="op">(</span>assign_by <span class="op">=</span> <span class="st">'razor-global'</span>, top_n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span>.method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'median'</span>, <span class="st">'linear'</span>, <span class="st">'loess'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pro_frompep_razor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="st">"sample"</span>, destination <span class="op">=</span> <span class="st">'return'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'sample'</span>, <span class="st">'proteins'</span>, <span class="st">'peptides'</span>, <span class="st">'CVs'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">sample</th>
<th align="right">proteins</th>
<th align="right">peptides</th>
<th align="right">CVs</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ctl</td>
<td align="right">7056</td>
<td align="right">63622</td>
<td align="right">0.14</td>
</tr>
<tr class="even">
<td align="left">p97_kd</td>
<td align="right">7056</td>
<td align="right">63622</td>
<td align="right">0.11</td>
</tr>
</tbody>
</table>
<p>Check to see how the two results agree â€¦</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl_merged_razor</span> <span class="op">&lt;-</span> <span class="va">pro_frompep_razor</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">'sample'</span>,<span class="st">'replicate'</span>,<span class="st">'protein'</span>, <span class="st">'abundance_raw'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span><span class="va">pro_data</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">'sample'</span>,<span class="st">'replicate'</span>,<span class="st">'protein'</span>, <span class="st">'abundance_raw'</span><span class="op">)</span>, </span>
<span>             by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'sample'</span>,<span class="st">'replicate'</span>,<span class="st">'protein'</span><span class="op">)</span>,</span>
<span>             suffix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"_tidyproteomics"</span>, <span class="st">"_proteomediscoverer"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tbl_merged_razor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>itentical_protein_abundances <span class="op">=</span> <span class="va">abundance_raw_proteomediscoverer</span> <span class="op">==</span> <span class="va">abundance_raw_tidyproteomics</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">itentical_protein_abundances</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">itentical_protein_abundances</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">'drop'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>r <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">n</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span>, <span class="st">"%"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">itentical_protein_abundances</th>
<th align="right">n</th>
<th align="left">r</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="right">812</td>
<td align="left">2.69257552143781 %</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="right">29345</td>
<td align="left">97.3074244785622 %</td>
</tr>
</tbody>
</table>
<p>A quick plot for visualizing the</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl_merged_razor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sample_rep <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">sample</span>, <span class="va">replicate</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">abundance_raw_proteomediscoverer</span>, <span class="va">abundance_raw_tidyproteomics</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">'red'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.67</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">sample_rep</span><span class="op">)</span></span></code></pre></div>
<p><img src="collapsing_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>Lets look at a few examples of outliers â€¦</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl_merged_razor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">abundance_raw_proteomediscoverer</span> <span class="op">!=</span> <span class="va">abundance_raw_tidyproteomics</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="8%">
<col width="11%">
<col width="9%">
<col width="33%">
<col width="37%">
</colgroup>
<thead><tr class="header">
<th align="left">sample</th>
<th align="left">replicate</th>
<th align="left">protein</th>
<th align="right">abundance_raw_tidyproteomics</th>
<th align="right">abundance_raw_proteomediscoverer</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ctl</td>
<td align="left">2</td>
<td align="left">P45983</td>
<td align="right">869555.1</td>
<td align="right">703721.1</td>
</tr>
<tr class="even">
<td align="left">ctl</td>
<td align="left">2</td>
<td align="left">Q8N138</td>
<td align="right">3530659.8</td>
<td align="right">2396096.2</td>
</tr>
<tr class="odd">
<td align="left">ctl</td>
<td align="left">2</td>
<td align="left">Q16537</td>
<td align="right">4131858.3</td>
<td align="right">3433024.4</td>
</tr>
</tbody>
</table>
<p>We will take the protein <strong>P45983</strong> to examine â€¦</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep_P45983</span> <span class="op">&lt;-</span> <span class="va">pep_data</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'p97_kd'</span>, <span class="va">replicate</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">protein</span> <span class="op">==</span> <span class="st">'P45983'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pep_data</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'p97_kd'</span>, <span class="va">replicate</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">peptide</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">pep_P45983</span><span class="op">$</span><span class="va">peptide</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">peptide</span>, <span class="va">modifications</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span></span>
<span>    proteins <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">protein</span>, collapse <span class="op">=</span> <span class="st">"; "</span><span class="op">)</span>,</span>
<span>    abundance_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="24%">
<col width="21%">
<col width="41%">
<col width="12%">
</colgroup>
<thead><tr class="header">
<th align="left">peptide</th>
<th align="left">modifications</th>
<th align="left">proteins</th>
<th align="right">abundance_raw</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">IIDFGLAR</td>
<td align="left">NA</td>
<td align="left">O15264; Q86YV6; P45984; P53778; Q16539; P45983</td>
<td align="right">3426127.8</td>
</tr>
<tr class="even">
<td align="left">ISVDEALQHPYINVWYDPSEAEAPPPK</td>
<td align="left">NA</td>
<td align="left">P45983</td>
<td align="right">417442.9</td>
</tr>
<tr class="odd">
<td align="left">MSYLLYQMLCGIK</td>
<td align="left">1xCarbamidomethyl [C10]</td>
<td align="left">P45984; P45983</td>
<td align="right">289726.1</td>
</tr>
<tr class="even">
<td align="left">VIEQLGTPCPEFMK</td>
<td align="left">1xCarbamidomethyl [C9]</td>
<td align="left">P45983</td>
<td align="right">566715.8</td>
</tr>
<tr class="odd">
<td align="left">YAGYSFEK</td>
<td align="left">NA</td>
<td align="left">P45983</td>
<td align="right">425560.5</td>
</tr>
</tbody>
</table>
<p>Here we see that <strong>P45983</strong> yielded 5 peptides, 2 of
which are shared. In the outlier table we see that ProteomeDiscoverer
excluded all razor peptides from the protein quantitation. And if we dig
further, we find that the peptide <strong>IIDFGLAR</strong> by razor
definition belongs to the protein <strong>Q16539</strong>, while the
peptide <strong>MSYLLYQMLCGIK</strong> can be assigned to
<strong>P45983</strong>, thus giving us the larger value shown in the
outlier table.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep_data</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'p97_kd'</span>, <span class="va">replicate</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">protein</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'O15264'</span>,<span class="st">'Q86YV6'</span>,<span class="st">'P45984'</span>,<span class="st">'P53778'</span>,<span class="st">'Q16539'</span>,<span class="st">'P45983'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">protein</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span></span>
<span>    n_peptides <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/length.html" class="external-link">length</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">peptide</span><span class="op">)</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/arrange.html" class="external-link">arrange</a></span><span class="op">(</span><span class="fu"><a href="https://dplyr.tidyverse.org/reference/desc.html" class="external-link">desc</a></span><span class="op">(</span><span class="va">n_peptides</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">protein</th>
<th align="right">n_peptides</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">Q16539</td>
<td align="right">7</td>
</tr>
<tr class="even">
<td align="left">P45983</td>
<td align="right">5</td>
</tr>
<tr class="odd">
<td align="left">P45984</td>
<td align="right">3</td>
</tr>
<tr class="even">
<td align="left">O15264</td>
<td align="right">2</td>
</tr>
<tr class="odd">
<td align="left">Q86YV6</td>
<td align="right">2</td>
</tr>
<tr class="even">
<td align="left">P53778</td>
<td align="right">1</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="assign-by-excluding-homologous-peptides">Assign by excluding homologous peptides<a class="anchor" aria-label="anchor" href="#assign-by-excluding-homologous-peptides"></a>
</h3>
<p>That means the ProteomeDiscoverer method for protein accounting
excluded the abundance measurement from all razor peptides. Let try to
do replicate that as well.</p>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pro_frompep_nonhom</span> <span class="op">&lt;-</span> <span class="va">pep_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/collapse.html">collapse</a></span><span class="op">(</span>assign_by <span class="op">=</span> <span class="st">'non-homologous'</span>, top_n <span class="op">=</span> <span class="cn">Inf</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span>.method <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'median'</span>, <span class="st">'linear'</span>, <span class="st">'loess'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pro_frompep_nonhom</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="st">"sample"</span>, destination <span class="op">=</span> <span class="st">'return'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'sample'</span>, <span class="st">'proteins'</span>, <span class="st">'peptides'</span>, <span class="st">'CVs'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">sample</th>
<th align="right">proteins</th>
<th align="right">peptides</th>
<th align="right">CVs</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ctl</td>
<td align="right">7056</td>
<td align="right">63622</td>
<td align="right">0.14</td>
</tr>
<tr class="even">
<td align="left">p97_kd</td>
<td align="right">7056</td>
<td align="right">63622</td>
<td align="right">0.12</td>
</tr>
</tbody>
</table>
<p>Check to see how the two results agree â€¦</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl_merged_nonhom</span> <span class="op">&lt;-</span> <span class="va">pro_frompep_nonhom</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">'sample'</span>,<span class="st">'replicate'</span>,<span class="st">'protein'</span>, <span class="st">'abundance_raw'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate-joins.html" class="external-link">inner_join</a></span><span class="op">(</span><span class="va">pro_data</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span><span class="st">'sample'</span>,<span class="st">'replicate'</span>,<span class="st">'protein'</span>, <span class="st">'abundance_raw'</span><span class="op">)</span>, </span>
<span>             by <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'sample'</span>,<span class="st">'replicate'</span>,<span class="st">'protein'</span><span class="op">)</span>,</span>
<span>             suffix <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">"_tidyproteomics"</span>, <span class="st">"_proteomediscoverer"</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">tbl_merged_nonhom</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>itentical_protein_abundances <span class="op">=</span> <span class="va">abundance_raw_proteomediscoverer</span> <span class="op">==</span> <span class="va">abundance_raw_tidyproteomics</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">itentical_protein_abundances</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">itentical_protein_abundances</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>n <span class="op">=</span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/context.html" class="external-link">n</a></span><span class="op">(</span><span class="op">)</span>, .groups <span class="op">=</span> <span class="st">'drop'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>r <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">n</span> <span class="op">/</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">n</span><span class="op">)</span> <span class="op">*</span> <span class="fl">100</span>, <span class="st">"%"</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="left">itentical_protein_abundances</th>
<th align="right">n</th>
<th align="left">r</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">FALSE</td>
<td align="right">3449</td>
<td align="left">11.5247101279781 %</td>
</tr>
<tr class="even">
<td align="left">TRUE</td>
<td align="right">26478</td>
<td align="left">88.4752898720219 %</td>
</tr>
</tbody>
</table>
<p>A quick plot for visualizing the</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl_merged_nonhom</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/mutate.html" class="external-link">mutate</a></span><span class="op">(</span>sample_rep <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">sample</span>, <span class="va">replicate</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/ggplot.html" class="external-link">ggplot</a></span><span class="op">(</span><span class="fu"><a href="https://ggplot2.tidyverse.org/reference/aes.html" class="external-link">aes</a></span><span class="op">(</span><span class="va">abundance_raw_proteomediscoverer</span>, <span class="va">abundance_raw_tidyproteomics</span><span class="op">)</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_abline.html" class="external-link">geom_abline</a></span><span class="op">(</span>color<span class="op">=</span><span class="st">'red'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/geom_point.html" class="external-link">geom_point</a></span><span class="op">(</span>alpha <span class="op">=</span> <span class="fl">.67</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_x_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/scale_continuous.html" class="external-link">scale_y_log10</a></span><span class="op">(</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/facet_wrap.html" class="external-link">facet_wrap</a></span><span class="op">(</span><span class="op">~</span><span class="va">sample_rep</span><span class="op">)</span></span></code></pre></div>
<p><img src="collapsing_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>Interesting, we now show that <strong>P45983</strong> agrees with
ProteomeDiscoverer, yet obviously several others do not</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl_merged_nonhom</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">protein</span> <span class="op">==</span> <span class="st">'P45983'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="8%">
<col width="11%">
<col width="9%">
<col width="33%">
<col width="37%">
</colgroup>
<thead><tr class="header">
<th align="left">sample</th>
<th align="left">replicate</th>
<th align="left">protein</th>
<th align="right">abundance_raw_tidyproteomics</th>
<th align="right">abundance_raw_proteomediscoverer</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ctl</td>
<td align="left">2</td>
<td align="left">P45983</td>
<td align="right">703721.1</td>
<td align="right">703721.1</td>
</tr>
<tr class="even">
<td align="left">p97_kd</td>
<td align="left">2</td>
<td align="left">P45983</td>
<td align="right">1635934.1</td>
<td align="right">1635934.1</td>
</tr>
<tr class="odd">
<td align="left">ctl</td>
<td align="left">1</td>
<td align="left">P45983</td>
<td align="right">535040.5</td>
<td align="right">535040.5</td>
</tr>
<tr class="even">
<td align="left">ctl</td>
<td align="left">3</td>
<td align="left">P45983</td>
<td align="right">494210.7</td>
<td align="right">494210.7</td>
</tr>
<tr class="odd">
<td align="left">p97_kd</td>
<td align="left">1</td>
<td align="left">P45983</td>
<td align="right">1409719.2</td>
<td align="right">1409719.2</td>
</tr>
<tr class="even">
<td align="left">p97_kd</td>
<td align="left">3</td>
<td align="left">P45983</td>
<td align="right">2219424.9</td>
<td align="right">2219424.9</td>
</tr>
</tbody>
</table>
<p>Lets look at a few examples of outliers â€¦</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl_merged_nonhom</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">abundance_raw_proteomediscoverer</span> <span class="op">!=</span> <span class="va">abundance_raw_tidyproteomics</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://rdrr.io/r/utils/head.html" class="external-link">head</a></span><span class="op">(</span><span class="fl">3</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="8%">
<col width="11%">
<col width="9%">
<col width="33%">
<col width="37%">
</colgroup>
<thead><tr class="header">
<th align="left">sample</th>
<th align="left">replicate</th>
<th align="left">protein</th>
<th align="right">abundance_raw_tidyproteomics</th>
<th align="right">abundance_raw_proteomediscoverer</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">ctl</td>
<td align="left">2</td>
<td align="left">Q15366</td>
<td align="right">67906.09</td>
<td align="right">112335311</td>
</tr>
<tr class="even">
<td align="left">ctl</td>
<td align="left">2</td>
<td align="left">Q9Y3L5</td>
<td align="right">81606.47</td>
<td align="right">4567305</td>
</tr>
<tr class="odd">
<td align="left">ctl</td>
<td align="left">2</td>
<td align="left">P53990</td>
<td align="right">114462.68</td>
<td align="right">16431834</td>
</tr>
</tbody>
</table>
<p>We will take the protein <strong>Q96LR5</strong> to examine â€¦</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep_Q96LR5</span> <span class="op">&lt;-</span> <span class="va">pep_data</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'p97_kd'</span>, <span class="va">replicate</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">protein</span> <span class="op">==</span> <span class="st">'Q96LR5'</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pep_data</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'p97_kd'</span>, <span class="va">replicate</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">peptide</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">pep_Q96LR5</span><span class="op">$</span><span class="va">peptide</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">peptide</span>, <span class="va">modifications</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span></span>
<span>    proteins <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">protein</span>, collapse <span class="op">=</span> <span class="st">"; "</span><span class="op">)</span>,</span>
<span>    abundance_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; `summarise()` has grouped output by 'peptide'. You can override using the</span></span>
<span><span class="co">#&gt; `.groups` argument.</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="22%">
<col width="33%">
<col width="27%">
<col width="16%">
</colgroup>
<thead><tr class="header">
<th align="left">peptide</th>
<th align="left">modifications</th>
<th align="left">proteins</th>
<th align="right">abundance_raw</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">DNWSPALTISK</td>
<td align="left">NA</td>
<td align="left">Q96LR5; P51965; Q969T4</td>
<td align="right">7005611.0</td>
</tr>
<tr class="even">
<td align="left">ELAEITLDPPPNCSAGPK</td>
<td align="left">1xCarbamidomethyl [C13]</td>
<td align="left">Q96LR5; Q969T4</td>
<td align="right">2169172.8</td>
</tr>
<tr class="odd">
<td align="left">IYHCNINSQGVICLDILK</td>
<td align="left">2xCarbamidomethyl [C4; C13]</td>
<td align="left">Q96LR5; P51965; Q969T4</td>
<td align="right">2543214.8</td>
</tr>
<tr class="even">
<td align="left">VDDSPSTSGGSSDGDQR</td>
<td align="left">NA</td>
<td align="left">Q96LR5</td>
<td align="right">216778.9</td>
</tr>
</tbody>
</table>
<p>A quick check on the peptide sum and it appears that
ProteomeDiscoverer ignored razor-peptides all together for this
protein</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">pep_data</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'p97_kd'</span>, <span class="va">replicate</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">peptide</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">pep_Q96LR5</span><span class="op">$</span><span class="va">peptide</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">peptide</span>, <span class="va">modifications</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span></span>
<span>    proteins <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">protein</span>, collapse <span class="op">=</span> <span class="st">"; "</span><span class="op">)</span>,</span>
<span>    abundance_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span>,</span>
<span>    .groups <span class="op">=</span> <span class="st">'drop'</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span>abundance_sum <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/sum.html" class="external-link">sum</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table class="table">
<thead><tr class="header">
<th align="right">abundance_sum</th>
</tr></thead>
<tbody><tr class="odd">
<td align="right">11934777</td>
</tr></tbody>
</table>
<p>If we examine the other proteins in that group for all the peptides
that may be shared â€¦</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">get_peptides</span> <span class="op">&lt;-</span> <span class="va">pep_data</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'p97_kd'</span>, <span class="va">replicate</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">protein</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Q96LR5'</span>,<span class="st">'P51965'</span>,<span class="st">'Q969T4'</span><span class="op">)</span><span class="op">)</span></span>
<span></span>
<span><span class="va">pep_data</span><span class="op">$</span><span class="va">quantitative</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="op">!</span><span class="fu"><a href="https://rdrr.io/r/base/NA.html" class="external-link">is.na</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'p97_kd'</span>, <span class="va">replicate</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">peptide</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="va">get_peptides</span><span class="op">$</span><span class="va">peptide</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/group_by.html" class="external-link">group_by</a></span><span class="op">(</span><span class="va">peptide</span>, <span class="va">modifications</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/summarise.html" class="external-link">summarise</a></span><span class="op">(</span></span>
<span>    proteins <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/paste.html" class="external-link">paste</a></span><span class="op">(</span><span class="va">protein</span>, collapse <span class="op">=</span> <span class="st">"; "</span><span class="op">)</span>,</span>
<span>    abundance_raw <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/unique.html" class="external-link">unique</a></span><span class="op">(</span><span class="va">abundance_raw</span><span class="op">)</span></span>
<span>  <span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; `summarise()` has grouped output by 'peptide'. You can override using the</span></span>
<span><span class="co">#&gt; `.groups` argument.</span></span></code></pre></div>
<table class="table">
<colgroup>
<col width="22%">
<col width="33%">
<col width="27%">
<col width="16%">
</colgroup>
<thead><tr class="header">
<th align="left">peptide</th>
<th align="left">modifications</th>
<th align="left">proteins</th>
<th align="right">abundance_raw</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">DNWSPALTISK</td>
<td align="left">NA</td>
<td align="left">Q96LR5; P51965; Q969T4</td>
<td align="right">7005611.0</td>
</tr>
<tr class="even">
<td align="left">DPAAPEPEEQEER</td>
<td align="left">NA</td>
<td align="left">Q969T4</td>
<td align="right">371158.4</td>
</tr>
<tr class="odd">
<td align="left">ELADITLDPPPNCSAGPK</td>
<td align="left">1xCarbamidomethyl [C13]</td>
<td align="left">P51965</td>
<td align="right">2536644.8</td>
</tr>
<tr class="even">
<td align="left">ELAEITLDPPPNCSAGPK</td>
<td align="left">1xCarbamidomethyl [C13]</td>
<td align="left">Q96LR5; Q969T4</td>
<td align="right">2169172.8</td>
</tr>
<tr class="odd">
<td align="left">IYHCNINSQGVICLDILK</td>
<td align="left">2xCarbamidomethyl [C4; C13]</td>
<td align="left">Q96LR5; P51965; Q969T4</td>
<td align="right">2543214.8</td>
</tr>
<tr class="even">
<td align="left">VDDSPSTSGGSSDGDQR</td>
<td align="left">NA</td>
<td align="left">Q96LR5</td>
<td align="right">216778.9</td>
</tr>
</tbody>
</table>
<p>â€¦ we see that ProteomeDiscoverer actually assigned the abundances of
razor-peptides. And if we check our results from the merged razor data
we find that to be correct.</p>
<div class="sourceCode" id="cb16"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">tbl_merged_razor</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'p97_kd'</span>, <span class="va">replicate</span> <span class="op">==</span> <span class="fl">1</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/filter.html" class="external-link">filter</a></span><span class="op">(</span><span class="va">protein</span> <span class="op"><a href="https://rdrr.io/r/base/match.html" class="external-link">%in%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'Q96LR5'</span>,<span class="st">'P51965'</span>,<span class="st">'Q969T4'</span><span class="op">)</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu">knitr</span><span class="fu">::</span><span class="fu"><a href="https://rdrr.io/pkg/knitr/man/kable.html" class="external-link">kable</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<table style="width:100%;" class="table">
<colgroup>
<col width="8%">
<col width="11%">
<col width="9%">
<col width="33%">
<col width="37%">
</colgroup>
<thead><tr class="header">
<th align="left">sample</th>
<th align="left">replicate</th>
<th align="left">protein</th>
<th align="right">abundance_raw_tidyproteomics</th>
<th align="right">abundance_raw_proteomediscoverer</th>
</tr></thead>
<tbody>
<tr class="odd">
<td align="left">p97_kd</td>
<td align="left">1</td>
<td align="left">Q969T4</td>
<td align="right">371158.4</td>
<td align="right">371158.4</td>
</tr>
<tr class="even">
<td align="left">p97_kd</td>
<td align="left">1</td>
<td align="left">P51965</td>
<td align="right">2536644.8</td>
<td align="right">2536644.8</td>
</tr>
<tr class="odd">
<td align="left">p97_kd</td>
<td align="left">1</td>
<td align="left">Q96LR5</td>
<td align="right">11934777.4</td>
<td align="right">11934777.4</td>
</tr>
</tbody>
</table>
</div>
<div class="section level3">
<h3 id="collapsing-to-gene-name">Collapsing to Gene Name<a class="anchor" aria-label="anchor" href="#collapsing-to-gene-name"></a>
</h3>
<p>Alternatively, peptide level data can be collapsed to any Annotation
term, in this case we will collapse to gene_name after adding the
annotations to the peptide data object. See
<code><a href="../articles/annotating.html">vignette("annotating")</a></code>.</p>
<div class="sourceCode" id="cb17"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://tidyr.tidyverse.org" class="external-link">tidyr</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_fasta</span> <span class="op">&lt;-</span> <span class="st">"uniprot_human-20398_20220920.fasta"</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu">tidyproteomics</span><span class="fu">:::</span><span class="fu"><a href="../reference/fasta_parse.html">fasta_parse</a></span><span class="op">(</span>as <span class="op">=</span> <span class="st">"data.frame"</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://dplyr.tidyverse.org/reference/select.html" class="external-link">select</a></span><span class="op">(</span>protein <span class="op">=</span> <span class="va">accession</span>, <span class="va">gene_name</span>, <span class="va">description</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="https://tidyr.tidyverse.org/reference/pivot_longer.html" class="external-link">pivot_longer</a></span><span class="op">(</span></span>
<span>    cols <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="st">'gene_name'</span>, <span class="st">'description'</span><span class="op">)</span>,</span>
<span>    names_to <span class="op">=</span> <span class="st">'term'</span>,</span>
<span>    values_to <span class="op">=</span> <span class="st">'annotation'</span></span>
<span>  <span class="op">)</span></span>
<span></span>
<span><span class="va">gene_data</span> <span class="op">&lt;-</span> <span class="va">pep_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/annotate.html">annotate</a></span><span class="op">(</span><span class="va">data_fasta</span>, duplicates <span class="op">=</span> <span class="st">'merge'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/collapse.html">collapse</a></span><span class="op">(</span>collapse_to <span class="op">=</span> <span class="st">'gene_name'</span><span class="op">)</span></span></code></pre></div>
<div class="sourceCode" id="cb18"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_counts.html">plot_counts</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span></code></pre></div>
<p><img src="collapsing_files/figure-html/unnamed-chunk-23-1.png" width="700"></p>
<div class="sourceCode" id="cb19"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">gene_data</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span>.method <span class="op">=</span> <span class="st">'linear'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/expression.html">expression</a></span><span class="op">(</span><span class="va">p97_kd</span><span class="op">/</span><span class="va">ctl</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">p97_kd</span><span class="op">/</span><span class="va">ctl</span>, significance_column <span class="op">=</span> <span class="st">'p_value'</span><span class="op">)</span> <span class="op">+</span> </span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"Gene Level Expression"</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Normalizing quantitative data</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> ... using <span style="color: #0000BB;">linear regression</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> ... using <span style="color: #0000BB;">linear regression</span> <span style="color: #B2B2B2;">[252ms]</span></span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span> Selecting best normalization method</span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">âœ”</span> Selecting best normalization method ... done</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â„¹</span>  ... selected <span style="color: #0000BB; font-style: italic;">linear</span></span></span>
<span><span class="co">#&gt; Warning: Values from `annotation` are not uniquely identified; output will contain</span></span>
<span><span class="co">#&gt; list-cols.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = list` to suppress this warning.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use `values_fn = {summary_fun}` to summarise duplicates.</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">â€¢</span> Use the following dplyr code to identify duplicates.</span></span>
<span><span class="co">#&gt;   {data} %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::group_by(gene_name, term) %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::summarise(n = dplyr::n(), .groups = "drop") %&gt;%</span></span>
<span><span class="co">#&gt;   dplyr::filter(n &gt; 1L)</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 82 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre></div>
<p><img src="collapsing_files/figure-html/unnamed-chunk-24-1.png" width="700"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jeff Jones.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
