<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="tidyproteomics: simplified data object and analysis functions
">
<title>Expression • tidyproteomics</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Expression">
<meta property="og:description" content="tidyproteomics: simplified data object and analysis functions
">
<meta property="og:image" content="https://jeffsocal.github.io/tidyproteomics/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">tidyproteomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.5.5</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/overview.html">Overview</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/updates.html">Updates</a>
    <a class="dropdown-item" href="../articles/data-object.html">The Data-Object</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Operations</h6>
    <a class="dropdown-item" href="../articles/importing.html">Importing</a>
    <a class="dropdown-item" href="../articles/summarizing.html">Summarizing</a>
    <a class="dropdown-item" href="../articles/subsetting.html">Subsetting</a>
    <a class="dropdown-item" href="../articles/normalizing.html">Normalizing</a>
    <a class="dropdown-item" href="../articles/imputing.html">Imputing</a>
    <a class="dropdown-item" href="../articles/annotating.html">Annotating</a>
    <a class="dropdown-item" href="../articles/collapsing.html">Protein Accounting</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Analysis</h6>
    <a class="dropdown-item" href="../articles/visualizing.html">Visualizing</a>
    <a class="dropdown-item" href="../articles/expression.html">Expression</a>
    <a class="dropdown-item" href="../articles/enrichment.html">Enrichment</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Workflows</h6>
    <a class="dropdown-item" href="../articles/workflow-simple.html">Simple</a>
    <a class="dropdown-item" href="../articles/workflow-importing.html">Importing Demo</a>
    <a class="dropdown-item" href="../articles/workflow-userimport.html">User Defined Import</a>
    <a class="dropdown-item" href="../articles/workflow-publication.html">Publication Demo</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Expression</h1>
            
      
      
      <div class="d-none name"><code>expression.Rmd</code></div>
    </div>

    
    
<div class="section level3">
<h3 id="analysis-of-two-samples">Analysis of Two Samples<a class="anchor" aria-label="anchor" href="#analysis-of-two-samples"></a>
</h3>
<p>One of the most fundamental analyses for quantitative proteomics is
the estimation of univariate differences between two samples.
Tidyproteomics accommodates two sample analysis using the
<code><a href="../reference/expression.html">expression()</a></code> function, the results of which can be
visualized using the <code><a href="../reference/plot_volcano.html">plot_volcano()</a></code> and
<code><a href="../reference/plot_proportion.html">plot_proportion()</a></code> visualization functions as shown below.
The expression analysis can be accomplished thought the R <em>limma</em>
package, or using any two-sample statistical comparison, the default
method employs the two-sided T-test. Note, if no values pass the cutoff
it will automatically choose the top proteins to display.</p>
<p>It should be noted that this package does not incorporate some of the
more recent methods described in literature. These methods could be
incorporated in future revisions, or through a user provided
function.</p>
<div class="section level4">
<h4 id="available-methods">Available Methods<a class="anchor" aria-label="anchor" href="#available-methods"></a>
</h4>
<table style="width:97%;" class="table">
<colgroup>
<col width="20%">
<col width="76%">
</colgroup>
<tbody>
<tr class="odd">
<td>t.test</td>
<td>
<p>Johnston, L. W. “Student’s t-Test.” <em>Journal of Quality
Technology</em> 2.4 (1970): 243-245.</p>
<p><a href="https://stat.eth%%2020z%20.ch/R-manual/R-devel/library/stats/html/t.test.html" class="external-link">R::stats::t.test</a></p>
</td>
</tr>
<tr class="even">
<td>
<p>limma</p>
<p>(Empirical Bayes)</p>
</td>
<td>
<p>Ritchie ME, Phipson B, Wu D, Hu Y, Law CW, Shi W, Smyth GK
(2015). “limma powers differential expression analyses for
RNA-sequencing and microarray studies.” <em>Nucleic Acids Research</em>,
<strong>43</strong>(7), e47.</p>
<p><a href="http://bi%%2020o%20conductor.org/packages/release/bioc/html/limma.html" class="external-link">Bioconductor:
limma</a></p>
</td>
</tr>
</tbody>
</table>
</div>
<div class="section level4">
<h4 id="alternative-methods">Alternative Methods<a class="anchor" aria-label="anchor" href="#alternative-methods"></a>
</h4>
<table style="width:94%;" class="table">
<colgroup>
<col width="20%">
<col width="73%">
</colgroup>
<tbody>
<tr class="odd">
<td>MSstats</td>
<td>
<p>Choi, Meena, Ching-Yun Chang, Timothy Clough, Daniel Broudy,
Trevor Killeen, Brendan MacLean, and Olga Vitek. 2014. “MSstats: An R
Package for Statistical Analysis of Quantitative Mass Spectrometry-Based
Proteomic Experiments.” <em>Bioinformatics</em> 30 (17): 2524–26.</p>
<p><a href="msstats.org">msstats.org</a></p>
</td>
</tr>
<tr class="even">
<td>DEqMS</td>
<td>
<p>Zhu, Yafeng, Lukas M. Orre, Yan Zhou Tran, Georgios Mermelekas,
Henrik J. Johansson, Alina Malyutina, Simon Anders, and Janne Lehtiö.
2020. “DEqMS: A Method for Accurate Variance Estimation in Differential
Protein Expression Analysis.” <em>Molecular &amp; Cellular Proteomics:
MCP</em> 19 (6): 1047–57.</p>
<p><a href="https://w%20ww.bi%20oco%20nductor.org/packages/release/bioc/html/DEqMS.html" class="external-link">Bioconductor:
DEqMS</a></p>
</td>
</tr>
<tr class="odd">
<td>MS -EmpiRe</td>
<td>
<p>Ammar, Constantin, Markus Gruber, Gergely Csaba, and Ralf Zimmer.
2019. “MS-EmpiRe Utilizes Peptide-Level Noise Distributions for
Ultra-Sensitive Detection of Differentially Expressed Proteins.”
<em>Molecular &amp; Cellular Proteomics: MCP</em> 18 (9): 1880–92.</p>
<p><a href="https://github.com/zimmerlab/MS-EmpiRe" class="external-link">GitH ub:
MS-EmpiRe</a></p>
</td>
</tr>
<tr class="even">
<td>meqrob</td>
<td>
<p>Sticker, Adriaan, Ludger Goeminne, Lennart Martens, and Lieven
Clement. 2020. “Robust Summarization and Inference in Proteome-Wide
Label-Free Quantification.” <em>Molecular &amp; Cellular Proteomics:
MCP</em> 19 (7): 1209–19.</p>
<p><a href="https://www%20.bioc%20ond%20uctor.org/packages/release/bioc/html/msqrob2.html" class="external-link">Bioconductor:
msqrob2</a></p>
</td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level3">
<h3 id="expression-analysis">Expression Analysis<a class="anchor" aria-label="anchor" href="#expression-analysis"></a>
</h3>
<div class="section level4">
<h4 id="input-parameters">Input Parameters<a class="anchor" aria-label="anchor" href="#input-parameters"></a>
</h4>
<table style="width:88%;" class="table">
<colgroup>
<col width="20%">
<col width="25%">
<col width="20%">
<col width="20%">
</colgroup>
<thead><tr class="header">
<th>parameter</th>
<th>description</th>
<th>inputs</th>
<th>default</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>log2fc_min</td>
<td>the minimum cutoff for displaying</td>
<td><em>0 to Inf</em></td>
<td><em>2</em></td>
</tr>
<tr class="even">
<td>signifi cance_column</td>
<td>the column to use for significance</td>
<td>
<ul>
<li>a</li>
</ul>
<p>dj_p_value,</p>
<p>p_value*</p>
</td>
<td><em>a dj_p_value</em></td>
</tr>
<tr class="odd">
<td>sign ificance_max</td>
<td>the minimum cutoff for displaying</td>
<td><em>1 to 1/Inf</em></td>
<td><em>0.05</em></td>
</tr>
<tr class="even">
<td>l abels_column</td>
<td>how to label the points</td>
<td>any value in annotations, eg <em>gene_name, description</em>
</td>
<td><em>protein</em></td>
</tr>
<tr class="odd">
<td>show_pannels</td>
<td>show the colored areas highlighting significance</td>
<td><em>TRUE / FALSE</em></td>
<td><em>TRUE</em></td>
</tr>
<tr class="even">
<td>show_lines</td>
<td>show the dotted cutoff lines</td>
<td><em>TRUE / FALSE</em></td>
<td><em>TRUE</em></td>
</tr>
<tr class="odd">
<td>s how_fc_scale</td>
<td>show the second scale on top converting <em>l og2_foldchange</em> to
<em>foldchange</em>
</td>
<td><em>TRUE / FALSE</em></td>
<td><em>TRUE</em></td>
</tr>
<tr class="even">
<td>point_size</td>
<td>size of each dot</td>
<td>
<em>1/10</em> to <em>5</em>
</td>
<td><em>NULL</em></td>
</tr>
<tr class="odd">
<td>co lor_positive</td>
<td>color for positive expression</td>
<td>[any R accepts] (htt%20ps:// r-charts%20. com/colors/)</td>
<td><em>dodgerblue</em></td>
</tr>
<tr class="even">
<td>co lor_negative</td>
<td>color for negative expression</td>
<td>[any R accepts] (htt%20ps:// r-charts%20. com/colors/)</td>
<td><em>firebrick1</em></td>
</tr>
</tbody>
</table>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://jeffsocal.github.io/tidyproteomics/">"tidyproteomics"</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">rdata</span> <span class="op">&lt;-</span> <span class="va">hela_proteins</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html" class="external-link">%&gt;%</a></span> </span>
<span>  <span class="fu"><a href="../reference/normalize.html">normalize</a></span><span class="op">(</span>.method <span class="op">=</span> <span class="st">'loess'</span><span class="op">)</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/expression.html">expression</a></span><span class="op">(</span><span class="va">knockdown</span><span class="op">/</span><span class="va">control</span><span class="op">)</span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="exporting-results">Exporting Results<a class="anchor" aria-label="anchor" href="#exporting-results"></a>
</h4>
<p>The results of this analysis are stored in the tidyproteomics
data-object and can easily be exported to save as a flat data table or
used in an external down stream analysis.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">rdata</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/export_analysis.html">export_analysis</a></span><span class="op">(</span><span class="va">knockdown</span><span class="op">/</span><span class="va">control</span>, .analysis <span class="op">=</span> <span class="st">'expression'</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># A tibble: 4,829 × 18</span></span></span>
<span><span class="co">#&gt;    protein imputed     n average_expression proportional_expression foldchange</span></span>
<span><span class="co">#&gt;    <span style="color: #949494; font-style: italic;">&lt;chr&gt;</span>     <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span> <span style="color: #949494; font-style: italic;">&lt;int&gt;</span>              <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>                   <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span>      <span style="color: #949494; font-style: italic;">&lt;dbl&gt;</span></span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 1</span> Q9NXV2    0         6           2<span style="text-decoration: underline;">768</span>778.              0.000<span style="text-decoration: underline;">011</span>3       14.0 </span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 2</span> Q8TBM8    0.333     6           3<span style="text-decoration: underline;">005</span>450.              0.000<span style="text-decoration: underline;">012</span>3        6.31</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 3</span> O95833    0.167     6           3<span style="text-decoration: underline;">047</span>900.              0.000<span style="text-decoration: underline;">012</span>4        6.02</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 4</span> Q14534    0.333     6           4<span style="text-decoration: underline;">043</span>660.              0.000<span style="text-decoration: underline;">016</span>5        4.98</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 5</span> Q5JUW8    0         6           1<span style="text-decoration: underline;">568</span>377.              0.000<span style="text-decoration: underline;">006</span>40       4.72</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 6</span> Q9Y385    0         6           1<span style="text-decoration: underline;">294</span>518.              0.000<span style="text-decoration: underline;">005</span>28       4.55</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 7</span> Q9Y4C2    0         6           6<span style="text-decoration: underline;">476</span>607.              0.000<span style="text-decoration: underline;">026</span>4        4.47</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 8</span> O00214    0.333     6           1<span style="text-decoration: underline;">257</span>127.              0.000<span style="text-decoration: underline;">005</span>13       4.38</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;"> 9</span> Q9H6H4    0         6           2<span style="text-decoration: underline;">167</span>884.              0.000<span style="text-decoration: underline;">008</span>84       4.25</span></span>
<span><span class="co">#&gt; <span style="color: #BCBCBC;">10</span> Q8WU67    0.333     6            <span style="text-decoration: underline;">928</span>918.              0.000<span style="text-decoration: underline;">003</span>79       4.10</span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 4,819 more rows</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;"># ℹ 12 more variables: log2_foldchange &lt;dbl&gt;, p_value &lt;dbl&gt;, adj_p_value &lt;dbl&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   description &lt;chr&gt;, gene_id_entrez &lt;chr&gt;, gene_id_ensemble &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   gene_name &lt;chr&gt;, biological_process &lt;chr&gt;, cellular_component &lt;chr&gt;,</span></span></span>
<span><span class="co">#&gt; <span style="color: #949494;">#   molecular_function &lt;chr&gt;, wiki_pathway &lt;chr&gt;, reactome_pathway &lt;chr&gt;</span></span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="volcano-plot">Volcano Plot<a class="anchor" aria-label="anchor" href="#volcano-plot"></a>
</h4>
<p>The default is to plot accoring to adjusted p-value with a Log2
fold-change cutoff of 2 or greater.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">rdata</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">knockdown</span><span class="op">/</span><span class="va">control</span><span class="op">)</span></span></code></pre></div>
<p><img src="expression_files/figure-html/unnamed-chunk-4-1.png" width="700"></p>
<p>The parameters can be adjusted to suit your needs …</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span></span>
<span><span class="va">rdata</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">knockdown</span><span class="op">/</span><span class="va">control</span>,</span>
<span>                       significance_column <span class="op">=</span> <span class="st">'p_value'</span>,</span>
<span>                       significance_max <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                       log2fc_min <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="expression_files/figure-html/unnamed-chunk-5-1.png" width="700"></p>
<p>… or desires.</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Attaching package: 'ggplot2'</span></span>
<span><span class="co">#&gt; The following object is masked from 'package:tidyproteomics':</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;     annotate</span></span>
<span></span>
<span><span class="va">rdata</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_volcano.html">plot_volcano</a></span><span class="op">(</span><span class="va">knockdown</span><span class="op">/</span><span class="va">control</span>,</span>
<span>                       significance_column <span class="op">=</span> <span class="st">'p_value'</span>,</span>
<span>                       significance_max <span class="op">=</span> <span class="fl">0.01</span>,</span>
<span>                       log2fc_min <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                       color_positive <span class="op">=</span> <span class="st">'orange'</span>,</span>
<span>                       color_negative <span class="op">=</span> <span class="st">'purple'</span>,</span>
<span>                       show_lines <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                       show_fc_scale <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                       show_pannels <span class="op">=</span> <span class="cn">FALSE</span>,</span>
<span>                       labels_column <span class="op">=</span> <span class="st">'gene_name'</span><span class="op">)</span> <span class="op">+</span></span>
<span>  <span class="fu"><a href="https://ggplot2.tidyverse.org/reference/labs.html" class="external-link">labs</a></span><span class="op">(</span>title <span class="op">=</span> <span class="st">"A nice volcano plot"</span>, subtitle <span class="op">=</span> <span class="st">"with great colors"</span><span class="op">)</span></span></code></pre></div>
<p><img src="expression_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="proportional-plot">Proportional Plot<a class="anchor" aria-label="anchor" href="#proportional-plot"></a>
</h4>
<p>The proportional plot is complementary to the volcano plot and
utilizes all the same underlying data. The default is to show both the
top 1% (<code>proportion_min = 0.01</code>) along with the values
passing the fold-change and significance cutoffs.</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">rdata</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_proportion.html">plot_proportion</a></span><span class="op">(</span><span class="va">knockdown</span><span class="op">/</span><span class="va">control</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning: ggrepel: 1 unlabeled data points (too many overlaps). Consider</span></span>
<span><span class="co">#&gt; increasing max.overlaps</span></span></code></pre></div>
<p><img src="expression_files/figure-html/unnamed-chunk-7-1.png" width="700"></p>
<p>This plot can likewise also be manipulated like the volcano plot.</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://ggplot2.tidyverse.org" class="external-link">ggplot2</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">rdata</span> <span class="op"><a href="https://rdrr.io/pkg/magrittr/man/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_proportion.html">plot_proportion</a></span><span class="op">(</span><span class="va">knockdown</span><span class="op">/</span><span class="va">control</span>,</span>
<span>                       significance_column <span class="op">=</span> <span class="st">'p_value'</span>,</span>
<span>                       proportion_min <span class="op">=</span> <span class="fl">0.1</span>,</span>
<span>                       log2fc_min <span class="op">=</span> <span class="fl">2</span>,</span>
<span>                       color_positive <span class="op">=</span> <span class="st">'orange'</span>,</span>
<span>                       color_negative <span class="op">=</span> <span class="st">'purple'</span>,</span>
<span>                       show_lines <span class="op">=</span> <span class="cn">TRUE</span>,</span>
<span>                       labels_column <span class="op">=</span> <span class="st">'gene_name'</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> <span style="color: #0000BB;">proportional_expression</span> appears to sum to 1 adjusting values to 100(%)</span></span></code></pre></div>
<p><img src="expression_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jeff Jones.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
