<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="tidyproteomics: simplified data object and analysis functions
">
<title>Summarizing • tidyproteomics</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Summarizing">
<meta property="og:description" content="tidyproteomics: simplified data object and analysis functions
">
<meta property="og:image" content="https://jeffsocal.github.io/tidyproteomics/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">tidyproteomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.0.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/get-started.html">Get Started</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/data-object.html">the tidyproteomics data-object</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Operations</h6>
    <a class="dropdown-item" href="../articles/importing.html">Importing</a>
    <a class="dropdown-item" href="../articles/summarizing.html">Summarizing</a>
    <a class="dropdown-item" href="../articles/subsetting.html">Subsetting</a>
    <a class="dropdown-item" href="../articles/normalizing.html">Normalizing</a>
    <a class="dropdown-item" href="../articles/imputing.html">Imputing</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Analysis</h6>
    <a class="dropdown-item" href="../articles/visualizing.html">Visualizing</a>
    <a class="dropdown-item" href="../articles/expression.html">Expression</a>
    <a class="dropdown-item" href="../articles/enrichment.html">Enrichment</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Workflows</h6>
    <a class="dropdown-item" href="../articles/workflow-simple.html">Simple</a>
    <a class="dropdown-item" href="../articles/workflow-publication.html">Publication Demo</a>
    <div class="dropdown-divider"></div>
    <a class="dropdown-item" href="../articles/updates.html">update history</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Summarizing</h1>
            
      
      
      <div class="d-none name"><code>summarizing.Rmd</code></div>
    </div>

    
    
<p>Summarizing proteomics data is vital to understanding the bigger
picture and conveying summary stats that set the tone for the larger
analysis. The results of each summary can be directed to via the
destination option to <code>"print"</code> on screen,
<code>"save"</code> to a file or <code>"return"</code> as a tibble.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://dplyr.tidyverse.org" class="external-link">"dplyr"</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="st"><a href="https://jeffsocal.github.io/tidyproteomics/">"tidyproteomics"</a></span><span class="op">)</span></span></code></pre></div>
<div class="section level3">
<h3 id="printing">Printing<a class="anchor" aria-label="anchor" href="#printing"></a>
</h3>
<p>The tidyproteomics data object can be printed to show a summary of
the object contents.</p>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold; font-style: italic;">Quantitative Proteomics Data Object</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Origin          ProteomeDiscoverer </span></span>
<span><span class="co">#&gt;                 proteins (4.13 MB) </span></span>
<span><span class="co">#&gt; Quantitation    8 files </span></span>
<span><span class="co">#&gt;                 2 samples (ko, wt) </span></span>
<span><span class="co">#&gt;                 2772 proteins </span></span>
<span><span class="co">#&gt;                 4.8 log10 dynamic range </span></span>
<span><span class="co">#&gt; Accounting      (4) match_between_runs num_peptides num_unique_peptides num_psms </span></span>
<span><span class="co">#&gt; Annotations     (9) description gene_id_entrez gene_id_ensemble gene_name</span></span>
<span><span class="co">#&gt;                 biological_process cellular_component</span></span>
<span><span class="co">#&gt;                 molecular_function wiki_pathway</span></span>
<span><span class="co">#&gt;                 reactome_pathway </span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p>And these can expand to encompass additions generated through other
operations.</p>
<div class="sourceCode" id="cb3"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/expression.html">expression</a></span><span class="op">(</span><span class="va">ko</span><span class="op">/</span><span class="va">wt</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/enrichment.html">enrichment</a></span><span class="op">(</span><span class="va">ko</span><span class="op">/</span><span class="va">wt</span>, .term <span class="op">=</span> <span class="st">'biological_process'</span><span class="op">)</span></span>
<span><span class="co">#&gt; Warning in preparePathwaysAndStats(pathways, stats, minSize, maxSize,</span></span>
<span><span class="co">#&gt; gseaParam, : There are duplicate gene names, fgsea may produce unexpected</span></span>
<span><span class="co">#&gt; results.</span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold; font-style: italic;">Quantitative Proteomics Data Object</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Origin          ProteomeDiscoverer </span></span>
<span><span class="co">#&gt;                 proteins (4.46 MB) </span></span>
<span><span class="co">#&gt; Quantitation    8 files </span></span>
<span><span class="co">#&gt;                 2 samples (ko, wt) </span></span>
<span><span class="co">#&gt;                 2772 proteins </span></span>
<span><span class="co">#&gt;                 4.8 log10 dynamic range </span></span>
<span><span class="co">#&gt; Accounting      (4) match_between_runs num_peptides num_unique_peptides num_psms </span></span>
<span><span class="co">#&gt; Annotations     (9) description gene_id_entrez gene_id_ensemble gene_name</span></span>
<span><span class="co">#&gt;                 biological_process cellular_component</span></span>
<span><span class="co">#&gt;                 molecular_function wiki_pathway</span></span>
<span><span class="co">#&gt;                 reactome_pathway </span></span>
<span><span class="co">#&gt; Analyses        (1) </span></span>
<span><span class="co">#&gt;                 ko/wt -&gt; expression &amp; enrichment (biological_process) </span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
</div>
<div class="section level3">
<h3 id="summarizing">Summarizing<a class="anchor" aria-label="anchor" href="#summarizing"></a>
</h3>
<div class="section level4">
<h4 id="groups">Groups<a class="anchor" aria-label="anchor" href="#groups"></a>
</h4>
<p>The tidyproteomics data object can summarize the quantitative and
accounting data …</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span><span class="op">)</span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Summary: global</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  files protein_groups proteins peptides peptides_unique quantifiable       CVs</span></span>
<span><span class="co">#&gt;      8           2772     2772    31271           31025         98.5 0.1573717</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p>… using columns corrisponing to experimental meta-data …</p>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span>by <span class="op">=</span> <span class="st">'sample'</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Summary: sample</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  sample files protein_groups proteins peptides peptides_unique quantifiable</span></span>
<span><span class="co">#&gt;      ko     4           2772     2772    31271           31025         98.4</span></span>
<span><span class="co">#&gt;      wt     4           2772     2772    31271           31025         98.5</span></span>
<span><span class="co">#&gt;        CVs</span></span>
<span><span class="co">#&gt;  0.1380689</span></span>
<span><span class="co">#&gt;  0.1612074</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p>… terms in the annotation meta-data …</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span>by <span class="op">=</span> <span class="st">'biological_process'</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Summary: biological_process</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;                biological_process files protein_groups proteins peptides</span></span>
<span><span class="co">#&gt;                     cell adhesion     8              1        1        1</span></span>
<span><span class="co">#&gt;  cell cycle OR cell proliferation     8              3        3       39</span></span>
<span><span class="co">#&gt;  cell organization and biogenesis     8             15       15      198</span></span>
<span><span class="co">#&gt;               cell-cell signaling     8           2772     2772    31271</span></span>
<span><span class="co">#&gt;           developmental processes     8              3        3        4</span></span>
<span><span class="co">#&gt;                    DNA metabolism     8           2772     2772    31271</span></span>
<span><span class="co">#&gt;        other biological processes     8            175      175     1262</span></span>
<span><span class="co">#&gt;         other metabolic processes     8            610      610     8233</span></span>
<span><span class="co">#&gt;                protein metabolism     8             18       18      332</span></span>
<span><span class="co">#&gt;   RNA metabolism OR transcription     8           2772     2772    31271</span></span>
<span><span class="co">#&gt;               signal transduction     8              1        1       18</span></span>
<span><span class="co">#&gt;                   stress response     8             45       45      330</span></span>
<span><span class="co">#&gt;                         transport     8            196      196     1542</span></span>
<span><span class="co">#&gt;  peptides_unique quantifiable       CVs</span></span>
<span><span class="co">#&gt;                1         62.5 0.2738303</span></span>
<span><span class="co">#&gt;               39        100.0 0.1084535</span></span>
<span><span class="co">#&gt;              197         98.4 0.1615593</span></span>
<span><span class="co">#&gt;            31025         98.5 0.1573717</span></span>
<span><span class="co">#&gt;                4         75.0 0.1980664</span></span>
<span><span class="co">#&gt;            31025         98.5 0.1573717</span></span>
<span><span class="co">#&gt;             1258         98.4 0.1752463</span></span>
<span><span class="co">#&gt;             8179         98.6 0.1511318</span></span>
<span><span class="co">#&gt;              332         99.9 0.1944947</span></span>
<span><span class="co">#&gt;            31025         98.5 0.1573717</span></span>
<span><span class="co">#&gt;               18        100.0 0.1228771</span></span>
<span><span class="co">#&gt;              316         93.6 0.1537054</span></span>
<span><span class="co">#&gt;             1529         97.6 0.1815972</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p>… and even terms in the accounting data …</p>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span>by <span class="op">=</span> <span class="st">'num_peptides'</span><span class="op">)</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Too many variables, limiting to the first 25</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Summary: num_peptides</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  num_peptides files protein_groups proteins peptides peptides_unique</span></span>
<span><span class="co">#&gt;            99     8              1        1       99              99</span></span>
<span><span class="co">#&gt;            91     8              1        1       91              91</span></span>
<span><span class="co">#&gt;            80     8              1        1       80              80</span></span>
<span><span class="co">#&gt;            75     8              1        1       75              75</span></span>
<span><span class="co">#&gt;            72     8              2        2      144             144</span></span>
<span><span class="co">#&gt;            68     8              1        1       68              68</span></span>
<span><span class="co">#&gt;            67     8              1        1       67              67</span></span>
<span><span class="co">#&gt;            65     8              1        1       65              65</span></span>
<span><span class="co">#&gt;            64     8              1        1       64              64</span></span>
<span><span class="co">#&gt;            62     8              1        1       62              56</span></span>
<span><span class="co">#&gt;            61     8              2        2      122             121</span></span>
<span><span class="co">#&gt;            60     8              3        3      180             180</span></span>
<span><span class="co">#&gt;            59     8              1        1       59              59</span></span>
<span><span class="co">#&gt;            58     8              1        1       58              58</span></span>
<span><span class="co">#&gt;            57     8              2        2      114             114</span></span>
<span><span class="co">#&gt;            56     8              3        3      168             168</span></span>
<span><span class="co">#&gt;            55     8              3        3      165             165</span></span>
<span><span class="co">#&gt;            54     8              1        1       54              54</span></span>
<span><span class="co">#&gt;            53     8              1        1       53              53</span></span>
<span><span class="co">#&gt;            51     8              5        5      255             255</span></span>
<span><span class="co">#&gt;            50     8              2        2      100              99</span></span>
<span><span class="co">#&gt;            49     8              4        4      196             196</span></span>
<span><span class="co">#&gt;            48     8              3        3      144             144</span></span>
<span><span class="co">#&gt;            46     8              2        2       92              92</span></span>
<span><span class="co">#&gt;            45     8              5        5      225             225</span></span>
<span><span class="co">#&gt;  quantifiable        CVs</span></span>
<span><span class="co">#&gt;         100.0 0.08083261</span></span>
<span><span class="co">#&gt;         100.0 0.09070565</span></span>
<span><span class="co">#&gt;         100.0 0.10035048</span></span>
<span><span class="co">#&gt;         100.0 0.25726125</span></span>
<span><span class="co">#&gt;         100.0 0.12178794</span></span>
<span><span class="co">#&gt;         100.0 0.07928306</span></span>
<span><span class="co">#&gt;         100.0 0.13334494</span></span>
<span><span class="co">#&gt;         100.0 0.08380745</span></span>
<span><span class="co">#&gt;         100.0 0.06669421</span></span>
<span><span class="co">#&gt;          90.3 0.07284085</span></span>
<span><span class="co">#&gt;          99.2 0.20278187</span></span>
<span><span class="co">#&gt;         100.0 0.11212260</span></span>
<span><span class="co">#&gt;         100.0 0.07984590</span></span>
<span><span class="co">#&gt;         100.0 0.18577896</span></span>
<span><span class="co">#&gt;         100.0 0.10527234</span></span>
<span><span class="co">#&gt;         100.0 0.07385101</span></span>
<span><span class="co">#&gt;         100.0 0.12369730</span></span>
<span><span class="co">#&gt;         100.0 0.08579622</span></span>
<span><span class="co">#&gt;         100.0 0.08548102</span></span>
<span><span class="co">#&gt;         100.0 0.12931300</span></span>
<span><span class="co">#&gt;          99.0 0.11365558</span></span>
<span><span class="co">#&gt;         100.0 0.08212217</span></span>
<span><span class="co">#&gt;         100.0 0.11745091</span></span>
<span><span class="co">#&gt;         100.0 0.12253259</span></span>
<span><span class="co">#&gt;         100.0 0.08962581</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<div class="sourceCode" id="cb8"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span>by <span class="op">=</span> <span class="st">'match_between_runs'</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Summary: match_between_runs</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  match_between_runs files protein_groups proteins peptides peptides_unique</span></span>
<span><span class="co">#&gt;               FALSE     8           2772     2772    30795           30576</span></span>
<span><span class="co">#&gt;                TRUE     8            527      527      477             450</span></span>
<span><span class="co">#&gt;  quantifiable       CVs</span></span>
<span><span class="co">#&gt;          99.1 0.1532708</span></span>
<span><span class="co">#&gt;          55.7 0.2856872</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
</div>
<div class="section level4">
<h4 id="contamination">Contamination<a class="anchor" aria-label="anchor" href="#contamination"></a>
</h4>
<p>In addition, the data can account for any type of contamination. If
the key word “CRAP” is use then the data will partition out by
<em>Keratin</em>, <em>BSA</em>, <em>Trypsin</em> and <em>Other</em>, so
long as the protein descriptions contain CRAP and presumably originate
from the <a href="https://reprint-apms.org/?q=chooseworkflow" class="external-link">CRAPome</a>. However,
any FASTA file, used at the data translation step (eg with
ProteomeDiscoverer), and manipulated to contain CRAP in the description
can be utilized.</p>
<div class="sourceCode" id="cb9"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span>contamination <span class="op">=</span> <span class="st">'CRAP'</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Summary: contamination</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  sample replicate native     BSA Keratin   Other  Trypsin sample_id</span></span>
<span><span class="co">#&gt;      ko         1  99.7%  0.103% 0.0946%  0.103% 0.00843%  f87cee7e</span></span>
<span><span class="co">#&gt;      ko         2  99.7%    0.1% 0.0924%    0.1% 0.00774%  fa3a5027</span></span>
<span><span class="co">#&gt;      ko         3  99.7% 0.0884% 0.0788% 0.0884% 0.00961%  fbf83a10</span></span>
<span><span class="co">#&gt;      ko         4  99.7%  0.102% 0.0951%  0.102% 0.00676%  feb72c95</span></span>
<span><span class="co">#&gt;      wt         1  99.7% 0.0861% 0.0779% 0.0861%  0.0082%  ff7546a2</span></span>
<span><span class="co">#&gt;      wt         2  99.7% 0.0858% 0.0801% 0.0858% 0.00577%  fd33f8fb</span></span>
<span><span class="co">#&gt;      wt         3  99.6%  0.117%  0.109%  0.117% 0.00786%  fcf192cc</span></span>
<span><span class="co">#&gt;      wt         4  99.6%  0.133%  0.125%  0.133% 0.00801%  f7add5f1</span></span>
<span><span class="co">#&gt;                  import_file sample_file</span></span>
<span><span class="co">#&gt;  pd_ecoli-hint_proteins.xlsx          f1</span></span>
<span><span class="co">#&gt;  pd_ecoli-hint_proteins.xlsx          f2</span></span>
<span><span class="co">#&gt;  pd_ecoli-hint_proteins.xlsx          f3</span></span>
<span><span class="co">#&gt;  pd_ecoli-hint_proteins.xlsx          f4</span></span>
<span><span class="co">#&gt;  pd_ecoli-hint_proteins.xlsx          f5</span></span>
<span><span class="co">#&gt;  pd_ecoli-hint_proteins.xlsx          f6</span></span>
<span><span class="co">#&gt;  pd_ecoli-hint_proteins.xlsx          f7</span></span>
<span><span class="co">#&gt;  pd_ecoli-hint_proteins.xlsx          f8</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p>Alternatively, any protein descriptor can also be use …</p>
<div class="sourceCode" id="cb10"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span>contamination <span class="op">=</span> <span class="st">'Trypsin'</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Summary: contamination</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  sample replicate native  Trypsin sample_id                 import_file</span></span>
<span><span class="co">#&gt;      ko         1   100% 0.00845%  f87cee7e pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      ko         2   100% 0.00775%  fa3a5027 pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      ko         3   100% 0.00963%  fbf83a10 pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      ko         4   100% 0.00678%  feb72c95 pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      wt         1   100% 0.00822%  ff7546a2 pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      wt         2   100% 0.00578%  fd33f8fb pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      wt         3   100% 0.00788%  fcf192cc pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      wt         4   100% 0.00803%  f7add5f1 pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;  sample_file</span></span>
<span><span class="co">#&gt;           f1</span></span>
<span><span class="co">#&gt;           f2</span></span>
<span><span class="co">#&gt;           f3</span></span>
<span><span class="co">#&gt;           f4</span></span>
<span><span class="co">#&gt;           f5</span></span>
<span><span class="co">#&gt;           f6</span></span>
<span><span class="co">#&gt;           f7</span></span>
<span><span class="co">#&gt;           f8</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p>… and as such.</p>
<div class="sourceCode" id="cb11"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="https://rdrr.io/r/base/summary.html" class="external-link">summary</a></span><span class="op">(</span>contamination <span class="op">=</span> <span class="st">'ribosome'</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold;">Summary: contamination</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt;  sample replicate native ribosome sample_id                 import_file</span></span>
<span><span class="co">#&gt;      ko         1  99.6%   0.389%  f87cee7e pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      ko         2  99.7%   0.349%  fa3a5027 pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      ko         3  99.6%   0.368%  fbf83a10 pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      ko         4  99.6%   0.363%  feb72c95 pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      wt         1  99.6%   0.356%  ff7546a2 pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      wt         2  99.6%   0.392%  fd33f8fb pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      wt         3  99.7%    0.35%  fcf192cc pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;      wt         4  99.6%    0.38%  f7add5f1 pd_ecoli-hint_proteins.xlsx</span></span>
<span><span class="co">#&gt;  sample_file</span></span>
<span><span class="co">#&gt;           f1</span></span>
<span><span class="co">#&gt;           f2</span></span>
<span><span class="co">#&gt;           f3</span></span>
<span><span class="co">#&gt;           f4</span></span>
<span><span class="co">#&gt;           f5</span></span>
<span><span class="co">#&gt;           f6</span></span>
<span><span class="co">#&gt;           f7</span></span>
<span><span class="co">#&gt;           f8</span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
</div>
</div>
<div class="section level3">
<h3 id="visualizing">Visualizing<a class="anchor" aria-label="anchor" href="#visualizing"></a>
</h3>
<p>Currently two summary visualizations are implemented in
tidyproteomics.</p>
<div class="section level4">
<h4 id="counts">Counts<a class="anchor" aria-label="anchor" href="#counts"></a>
</h4>
<p>In a simple grouped barchart, proteins (individual, and groups) as
well as peptides (all, unique) are displayed with the match-between-runs
shown as the margin above the ms2 evidenced identifications.</p>
<div class="sourceCode" id="cb12"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_counts.html">plot_counts</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="summarizing_files/figure-html/unnamed-chunk-13-1.png" width="700"></p>
</div>
<div class="section level4">
<h4 id="quantitation">Quantitation<a class="anchor" aria-label="anchor" href="#quantitation"></a>
</h4>
<p>In recent literature a summary of protein quantitation has been
visualized as a rank-based dot-plot.</p>
<div class="sourceCode" id="cb13"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_quantrank.html">plot_quantrank</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="summarizing_files/figure-html/unnamed-chunk-14-1.png" width="700"></p>
<p>Additionally, this plot can be extended to highlight statistical
differences via an unbiased all-pair-wise comparison, to give an
anticipated view of how to guide downstream analyses.</p>
<div class="sourceCode" id="cb14"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_quantrank.html">plot_quantrank</a></span><span class="op">(</span>display_filter <span class="op">=</span> <span class="st">'log2_foldchange'</span>, display_cutoff <span class="op">=</span> <span class="fl">2</span><span class="op">)</span></span></code></pre></div>
<p><img src="summarizing_files/figure-html/unnamed-chunk-15-1.png" width="700"></p>
<p>Filter to limit to the range <code>c(low, high)</code> and display
the protein ids.</p>
<div class="sourceCode" id="cb15"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">ecoli_proteins</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_quantrank.html">plot_quantrank</a></span><span class="op">(</span>show_rank_scale <span class="op">=</span> <span class="cn">TRUE</span>, limit_rank <span class="op">=</span> <span class="fu"><a href="https://rdrr.io/r/base/c.html" class="external-link">c</a></span><span class="op">(</span><span class="fl">1</span>,<span class="fl">25</span><span class="op">)</span><span class="op">)</span></span></code></pre></div>
<p><img src="summarizing_files/figure-html/unnamed-chunk-16-1.png" width="700"></p>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jeff Jones.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
