<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="tidyproteomics: simplified data object and analysis functions
">
<title>Update History • tidyproteomics</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.3.1/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.3.1/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Update History">
<meta property="og:description" content="tidyproteomics: simplified data object and analysis functions
">
<meta property="og:image" content="https://jeffsocal.github.io/tidyproteomics/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">tidyproteomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.8.8</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/overview.html">Overview</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/updates.html">Updates</a>
    <a class="dropdown-item" href="../articles/data-object.html">The Data-Object</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Basics</h6>
    <a class="dropdown-item" href="../articles/importing.html">Importing</a>
    <a class="dropdown-item" href="../articles/annotating.html">Annotating</a>
    <a class="dropdown-item" href="../articles/summarizing.html">Summarizing</a>
    <a class="dropdown-item" href="../articles/visualizing.html">Visualizing</a>
    <a class="dropdown-item" href="../articles/extracting.html">Extracting</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Processing</h6>
    <a class="dropdown-item" href="../articles/subsetting.html">Subsetting</a>
    <a class="dropdown-item" href="../articles/normalizing.html">Normalizing</a>
    <a class="dropdown-item" href="../articles/imputing.html">Imputing</a>
    <a class="dropdown-item" href="../articles/automating.html">Automating</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Analysis</h6>
    <a class="dropdown-item" href="../articles/expression.html">Expression</a>
    <a class="dropdown-item" href="../articles/enrichment.html">Enrichment</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Peptides to Proteins</h6>
    <a class="dropdown-item" href="../articles/collapsing.html">Collapsing</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Example Workflows</h6>
    <a class="dropdown-item" href="../articles/workflow-simple.html">Simple</a>
    <a class="dropdown-item" href="../articles/workflow-importing.html">Importing Demo</a>
    <a class="dropdown-item" href="../articles/workflow-publication.html">Publication Demo</a>
    <a class="dropdown-item" href="../articles/workflow-userimport.html">User Defined Import</a>
  </div>
</li>
<li class="nav-item">
  <a class="nav-link" href="../articles/updates.html">Updates</a>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Update History</h1>
            
      
      
      <div class="d-none name"><code>updates.Rmd</code></div>
    </div>

    
    
<p>Versioning follows a semantic scheme with a brief accounting of the
major updates and bug-fixes listed here. For quick reference:</p>
<ul>
<li><p>x.x.N - minor updates or bug fixes</p></li>
<li><p>x.N.x - introduction of new features or improvements that should
not break existing implementations</p></li>
<li><p>N.x.x - major updates or changes that would break backward
compatibility</p></li>
</ul>
<hr>
<div class="section level3">
<h3 id="version-1-8-x">Version: 1.8.x<a class="anchor" aria-label="anchor" href="#version-1-8-x"></a>
</h3>
<div class="section level4">
<h4 id="section">1.8.8 - 2025.06.10<a class="anchor" aria-label="anchor" href="#section"></a>
</h4>
<ul>
<li>Updated volcano plots to show only whole values on x-axis.</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-1">1.8.7 - 2025.05.29<a class="anchor" aria-label="anchor" href="#section-1"></a>
</h4>
<ul>
<li>Minor fix to collapse() resolving an printing of debug
information.</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-2">1.8.6 - 2025.05.06<a class="anchor" aria-label="anchor" href="#section-2"></a>
</h4>
<ul>
<li>Minor fix to collapse() [thanks: github.com/rzieg]. Resolved an
issue where selecting “all-possible” could result in only a single
protein being chosen when two proteins shared identical peptide
sequences. This situation can occur when searching against two FASTA
databases that contain nearly identical proteins differing by only a
single amino acid—such as between cRAP and Human sequence databases.
Typically, these sequences are identical and don’t cause issues, but
this edge case led to unintended behavior.</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-3">1.8.5 - 2024.12.18<a class="anchor" aria-label="anchor" href="#section-3"></a>
</h4>
<ul>
<li>Minor fix to <code><a href="../reference/normalize.html">normalize()</a></code> [thanks: github.com/tywang-tw]
where after limma NA are reintroduced into the quantitative data table
compromising later data exports.</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-4">1.8.4 - 2024.10.14<a class="anchor" aria-label="anchor" href="#section-4"></a>
</h4>
<ul>
<li>Minor fix to <code>merge() |&gt; export_quant()</code> [thanks:
github.com/tywang-tw] where the quantitative source ‘selected’ was not
recognized. This has been addressed, however, it is un-advised to
<code><a href="../reference/normalize.html">normalize()</a></code> prior to <code><a href="../reference/merge.html">merge()</a></code> as the
possibility exists to have different normalization methods used between
sets and major quantitative offsets, that normalization is intended to
address, persist or become exaggerated, increasing the probability of
false positives in downstream analyses.</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-5">1.8.3 - 2024.09.24<a class="anchor" aria-label="anchor" href="#section-5"></a>
</h4>
<ul>
<li>Implemented additional columns on to calculate the fraction imputed
per sample when exporting to a flat table [thanks:
github.com/tywang-tw].</li>
<li>Bug fix on peptide-to-protein collapse in some cases resulting in
expanded sample sizes due to mismatch on table merge.</li>
<li>Bug fix to account for enrichment estimates when protein log2 fold
change values are significantly skewed.</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-6">1.8.2 - 2024.08.02<a class="anchor" aria-label="anchor" href="#section-6"></a>
</h4>
<ul>
<li>Updated the volcano plot to include auto-generated titles, classic
theme, plotting only y=p-value with either p-value or adj-p-value
significance cutoffs and removing values with p-value==1.</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-7">1.8.1 - 2024.07.30<a class="anchor" aria-label="anchor" href="#section-7"></a>
</h4>
<ul>
<li>Added Fisher’s Exact Test (Over-Representation Analysis) to
enrichment() function.</li>
<li>Added the ability to automate both expression() and enrichment()
functions with updated .terms parameter</li>
<li>Added the analyze_expressions() and analyze_enrichments() to
compliment the automation by saving out plots and tables for each
expression and enrichment test.</li>
</ul>
<hr>
</div>
</div>
<div class="section level3">
<h3 id="version-1-7-x">Version: 1.7.x<a class="anchor" aria-label="anchor" href="#version-1-7-x"></a>
</h3>
<div class="section level4">
<h4 id="section-8">1.7.4 - 2024.06.04<a class="anchor" aria-label="anchor" href="#section-8"></a>
</h4>
<ul>
<li>Bug fix missing package reference in limma expression analysis</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-9">1.7.3 - 2024.03.12<a class="anchor" aria-label="anchor" href="#section-9"></a>
</h4>
<ul>
<li>fixed issue with importing FragPipe tsv files, seems to be an issue
with the <code>vroom</code> package. Modified to use the
<code>readr</code> package for data importing.</li>
<li>modified the <code><a href="../reference/export_quant.html">export_quant()</a></code> function to allow exporting
proteins with NAs for all samples</li>
<li>changed <code><a href="https://rdrr.io/r/base/subset.html" class="external-link">subset()</a></code> to be an S3 function</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-10">1.7.2 - 2024.01.24<a class="anchor" aria-label="anchor" href="#section-10"></a>
</h4>
<ul>
<li>fixed and issue with plot_counts and plot_variation_cv failed to
account for sample sets larger than 16 when providing a color
palette</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-11">1.7.1 - 2024.01.23<a class="anchor" aria-label="anchor" href="#section-11"></a>
</h4>
<ul>
<li>new release to add DOI from Zenodo</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-12">1.7.0 - 2023.12.13<a class="anchor" aria-label="anchor" href="#section-12"></a>
</h4>
<ul>
<li>add the ability to export the proteins specific to a given Venn
intersection <a href="./extracting.html">extracting</a>
</li>
<li>minor improvements on the export_analysis function to clarify column
headers</li>
</ul>
<hr>
</div>
</div>
<div class="section level3">
<h3 id="version-1-6-x">Version: 1.6.x<a class="anchor" aria-label="anchor" href="#version-1-6-x"></a>
</h3>
<div class="section level4">
<h4 id="section-13">1.6.1 - 2023.11.29<a class="anchor" aria-label="anchor" href="#section-13"></a>
</h4>
<ul>
<li>fixed some minor issues</li>
<li>implemented the request from issue#9 adding the quantitative values
to the export_analysis() function</li>
<li>implemented the request from issue#12 allowing for the iterative
evaluation of multiple comparisons for the expression() and enrichment()
functions</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-14">1.6.0 - 2023.11.09<a class="anchor" aria-label="anchor" href="#section-14"></a>
</h4>
<ul>
<li>fixed an issue with improper missing value accounting when
implementing expression tests [thanks: github.com/rzieg]</li>
</ul>
<hr>
</div>
</div>
<div class="section level3">
<h3 id="version-1-5-x">Version: 1.5.x<a class="anchor" aria-label="anchor" href="#version-1-5-x"></a>
</h3>
<div class="section level4">
<h4 id="section-15">1.5.10 - 2023.11.03<a class="anchor" aria-label="anchor" href="#section-15"></a>
</h4>
<ul>
<li>addressed issues#9 &amp; #10 to better handle as.data.frame [thanks:
github.com/tywang-tw]</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-16">1.5.9 - 2023.10.30<a class="anchor" aria-label="anchor" href="#section-16"></a>
</h4>
<ul>
<li>addressed issue#8 to better handle the text of custom functions
[thanks: github.com/michwij]</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-17">1.5.8 - 2023.10.10<a class="anchor" aria-label="anchor" href="#section-17"></a>
</h4>
<ul>
<li>addressed issue#5 with the default MaxQuant protein import
definition [thanks: tar226-at-cornell.edu]</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-18">1.5.7 - 2023.09.21<a class="anchor" aria-label="anchor" href="#section-18"></a>
</h4>
<ul>
<li>addressed issue#4 with enrichment() taking long for complex
annotations, implemented parallel::mclapply [thanks:
github.com/eschen42]</li>
<li>created a export_config() function to assist in custom import
definitions [thanks: github.com/eschen42]</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-19">1.5.6 - 2023.09.07<a class="anchor" aria-label="anchor" href="#section-19"></a>
</h4>
<ul>
<li>addressed issue#3: t.test on expression diff fails on no variance
data</li>
<li>fixed issue with reporting imputation in object details and record
of operations</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-20">1.5.5 - 2023.08.24<a class="anchor" aria-label="anchor" href="#section-20"></a>
</h4>
<ul>
<li>fixed deprecated issue in subset (tidyr::pivot_wider)</li>
<li>fixed issue with use R v4+ pipe operator</li>
<li>other minor improvements</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-21">1.5.4 - 2023.08.16<a class="anchor" aria-label="anchor" href="#section-21"></a>
</h4>
<ul>
<li>added support for selecting fgsea scoreType(“std”, “pos”, “neg”)
[thanks: github.com/tywang-tw]</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-22">1.5.3 - 2023.07.30<a class="anchor" aria-label="anchor" href="#section-22"></a>
</h4>
<ul>
<li>added theme color support for plotting 16+ sample groups</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-23">1.5.2 - 2023.06.29<a class="anchor" aria-label="anchor" href="#section-23"></a>
</h4>
<ul>
<li>bug fix in limma expression differences (name ordering only
effecting data with 3+ sample groups) [thanks:
github.com/tywang-tw]</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-24">1.5.1 - 2023.06.09<a class="anchor" aria-label="anchor" href="#section-24"></a>
</h4>
<ul>
<li>updated plot_counts(), added show_replicates = T/F</li>
</ul>
</div>
<div class="section level4">
<h4 id="section-25">1.5.0 - 2023.05.23<a class="anchor" aria-label="anchor" href="#section-25"></a>
</h4>
<ul>
<li>refactored impute() to allow for user defined matrix based
functions</li>
<li>refactored reassign() to use expressive comparisons</li>
<li>added export_compexp() to export the data under the two-expression
plot</li>
<li>updated plot_variation_cv() to better label y-axis</li>
<li>modified summary() for peptides using simpler accounting</li>
<li>bugfix on import() allowing &gt; 0 annotation columns</li>
<li>bugfix in summary() with table merging</li>
<li>bugfix in annotate() with leftover dplyr groups</li>
</ul>
<hr>
</div>
</div>
<div class="section level3">
<h3 id="version-1-4-x">Version: 1.4.x<a class="anchor" aria-label="anchor" href="#version-1-4-x"></a>
</h3>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-26">1.4.2 - 2023.04.03<a class="anchor" aria-label="anchor" href="#section-26"></a>
</h4>
<ul>
<li>updated color theme to maximize contrasts in order</li>
<li>updated plot_quantrank() to further fade background points when
using a display_subset</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-27">1.4.1 - 2023.03.28<a class="anchor" aria-label="anchor" href="#section-27"></a>
</h4>
<ul>
<li>migrated from dplyr deprecated across() to group_by_at()</li>
<li>minor bugfix on accounting by non-protein based values</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-28">1.4.0 - 2023.03.27<a class="anchor" aria-label="anchor" href="#section-28"></a>
</h4>
<ul>
<li>added the S3Method as.data.frame()</li>
</ul>
<hr>
</div>
</div>
<div class="section level3">
<h3 id="version-1-3-x">Version: 1.3.x<a class="anchor" aria-label="anchor" href="#version-1-3-x"></a>
</h3>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-29">1.3.1 - 2023.03.20<a class="anchor" aria-label="anchor" href="#section-29"></a>
</h4>
<ul>
<li>minor code updates to improve conciseness</li>
<li>added display_subset to plot_quantrank() to highlight user selected
proteins/peptides amoung the whole dynamic range</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-30">1.3.0 - 2023.03.13<a class="anchor" aria-label="anchor" href="#section-30"></a>
</h4>
<ul>
<li>addressed multiple charge precursor in peptide quant files</li>
<li>added FragPipe to the available imports</li>
<li>exposed adjust.p.value method selection in expression()</li>
<li>exposed the user defined import option</li>
<li>fixed a labeling error in plot_pca()</li>
<li>modified the method for pivoting columns-to-rows in import
files</li>
</ul>
<hr>
</div>
</div>
<div class="section level3">
<h3 id="version-1-2-x">Version: 1.2.x<a class="anchor" aria-label="anchor" href="#version-1-2-x"></a>
</h3>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-31">1.2.6 - 2023.03.01<a class="anchor" aria-label="anchor" href="#section-31"></a>
</h4>
<ul>
<li>added clarity around normalize(), impute(), collapse()</li>
<li>implemented parallel processing for ‘randomforest’ and ‘svm’ in
normalize()</li>
<li>added a vignette on protein inferencing using collapse()</li>
<li>fixed a bug in peptide importing that removed homologous peptide
accounting</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-32">1.2.5 - 2023.02.13<a class="anchor" aria-label="anchor" href="#section-32"></a>
</h4>
<ul>
<li>fixed an issue with MBR accounting</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-33">1.2.4 - 2023.02.03<a class="anchor" aria-label="anchor" href="#section-33"></a>
</h4>
<ul>
<li>added support for mzTab v1.0.0, exporting table for counts
analysis</li>
</ul>
<hr>
</div>
</div>
<div class="section level3">
<h3 id="version-1-1-x">Version: 1.1.x<a class="anchor" aria-label="anchor" href="#version-1-1-x"></a>
</h3>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-34">1.1.2 - 2023.01.23<a class="anchor" aria-label="anchor" href="#section-34"></a>
</h4>
<ul>
<li>refactored the import function to allow both wide and long format
data in a configuration file, eliminating the hard coded import
functions for ProteomeDiscoverer and MaxQuant</li>
</ul>
</div>
</div>
<div class="section level3">
<h3 id="version-1-0-x">Version: 1.0.x<a class="anchor" aria-label="anchor" href="#version-1-0-x"></a>
</h3>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-35">1.0.4 - 2022.11.28<a class="anchor" aria-label="anchor" href="#section-35"></a>
</h4>
<ul>
<li>modified %!like% to the more conventional !x %like% y</li>
<li>added the rm.mbr() function</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-36">1.0.2 - 2022.11.28<a class="anchor" aria-label="anchor" href="#section-36"></a>
</h4>
<ul>
<li>implemented S3 object for tidyproteomics data object</li>
<li>refactored all functions to accout for new data object</li>
<li>fixed minor issues with exporting data and plots</li>
<li>updated and added more vignettes</li>
</ul>
<hr>
</div>
</div>
<div class="section level3">
<h3 id="version-beta">Version: beta<a class="anchor" aria-label="anchor" href="#version-beta"></a>
</h3>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-37">0.7.3 - 2022.11.15<a class="anchor" aria-label="anchor" href="#section-37"></a>
</h4>
<ul>
<li>modified expression() to more accurately account for the imputation
ratio</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-38">0.7.2 - 2022.11.03<a class="anchor" aria-label="anchor" href="#section-38"></a>
</h4>
<ul>
<li>refactored plotting to utilize a generalized save function to
facilitate pipelining</li>
<li>modified subset to behave like a tidyverse function while also
introducing the operators <code>like</code> and <code>!like</code> to
subset data based on a string matching comparison</li>
<li>updated the functions normalize(), plot_counts() and summary() that
utilize subset()</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-39">0.7.1 - 2022.10.28<a class="anchor" aria-label="anchor" href="#section-39"></a>
</h4>
<ul>
<li>refactored import() to allow for user based configurations</li>
<li>refactored subset(), eliminating remove(), to account for value
comparison and pattern matching</li>
<li>refactored reassign(), from modify(), to improve sample
renaming</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-40">0.7.0 - 2022.10.15<a class="anchor" aria-label="anchor" href="#section-40"></a>
</h4>
<ul>
<li>added plot_quantrank()</li>
<li>fixed issue with ProteomeDiscoverer importing of TMT data</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-41">0.6.0 - 2022.09.29<a class="anchor" aria-label="anchor" href="#section-41"></a>
</h4>
<ul>
<li>minor bug fixes</li>
<li>refactored rollup() to collapse()</li>
<li>defined meld() and codify() to create uniformity in data import and
segmentation</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-42">0.5.0 - 2022.09.23<a class="anchor" aria-label="anchor" href="#section-42"></a>
</h4>
<ul>
<li>created remove() and migrated subset(match_between_runs =
FALSE)</li>
<li>improved protein accounting in import() and rollup() to include
protein group accounting</li>
<li>added plot_counts()</li>
<li>added import(‘Skyline’, ‘peptides’), Skyline can handel DDA, PRM via
MSAmanda and DIA vi DIA-Umpire</li>
<li>added import(‘DIA-NN’, ‘peptides’), DIA-NN can handel DIA</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-43">0.4.2 - 2022.09.19<a class="anchor" aria-label="anchor" href="#section-43"></a>
</h4>
<ul>
<li>fixed issue with “out of memory” on subset(match_between_runs =
FALSE)</li>
<li>fixed an x-axis ordering bug plot_variation()</li>
<li>fixed a minor bug in importing ProteomeDiscoverer peptide data</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-44">0.4.1 - 2022.09.13<a class="anchor" aria-label="anchor" href="#section-44"></a>
</h4>
<ul>
<li>fixed a minor bug in importing MaxQuant data (empty data
object)</li>
<li>fixed a minor bug in plot_protein()</li>
<li>fixed bug in annotations()</li>
<li>fixed bug in impute() with randomforest</li>
<li>fixed bug in expression() with limma</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-45">0.4.0 - 2022.09.08<a class="anchor" aria-label="anchor" href="#section-45"></a>
</h4>
<ul>
<li>added import() for MaxQuant peptides (evidence.txt)</li>
<li>added import() for MaxQuant proteins (proteinGroups.txt)</li>
<li>updated import() for ProteomeDiscoverer (better configs)</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-46">0.3.0<a class="anchor" aria-label="anchor" href="#section-46"></a>
</h4>
<ul>
<li>added protein rollup() function, integrated with <a href="https://github.com/jeffsocal/rfasta" class="external-link">rfasta</a>
</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-47">0.2.0<a class="anchor" aria-label="anchor" href="#section-47"></a>
</h4>
<ul>
<li>updated impute() accounting</li>
<li>added plot_enrichment()</li>
<li>added history of data transformations via operations()</li>
</ul>
</div>
<div class="section level4 unnumbered">
<h4 class="unnumbered" id="section-48">0.1.0<a class="anchor" aria-label="anchor" href="#section-48"></a>
</h4>
<ul>
<li>initial cohesion of a loose collection of functions into an R
package</li>
</ul>
</div>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jeff Jones.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
