<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en">
<head>
<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<meta charset="utf-8">
<meta http-equiv="X-UA-Compatible" content="IE=edge">
<meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<meta name="description" content="tidyproteomics: simplified data object and analysis functions
">
<title>User Defined Import • tidyproteomics</title>
<!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png">
<link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png">
<link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png">
<link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png">
<link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png">
<link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png">
<script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">
<link href="../deps/bootstrap-5.2.2/bootstrap.min.css" rel="stylesheet">
<script src="../deps/bootstrap-5.2.2/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous">
<link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous">
<!-- bootstrap-toc --><script src="https://cdn.jsdelivr.net/gh/afeld/bootstrap-toc@v1.0.1/dist/bootstrap-toc.min.js" integrity="sha256-4veVQbu7//Lk5TSmc7YV48MxtMy98e26cf5MrgZYnwo=" crossorigin="anonymous"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="User Defined Import">
<meta property="og:description" content="tidyproteomics: simplified data object and analysis functions
">
<meta property="og:image" content="https://jeffsocal.github.io/tidyproteomics/logo.png">
<!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]-->
</head>
<body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">tidyproteomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.7.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto">
<li class="nav-item">
  <a class="nav-link" href="../articles/overview.html">Overview</a>
</li>
<li class="nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="active nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/updates.html">Updates</a>
    <a class="dropdown-item" href="../articles/data-object.html">The Data-Object</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Operations</h6>
    <a class="dropdown-item" href="../articles/importing.html">Importing</a>
    <a class="dropdown-item" href="../articles/summarizing.html">Summarizing</a>
    <a class="dropdown-item" href="../articles/subsetting.html">Subsetting</a>
    <a class="dropdown-item" href="../articles/normalizing.html">Normalizing</a>
    <a class="dropdown-item" href="../articles/imputing.html">Imputing</a>
    <a class="dropdown-item" href="../articles/annotating.html">Annotating</a>
    <a class="dropdown-item" href="../articles/collapsing.html">Protein Accounting</a>
    <a class="dropdown-item" href="../articles/extracting.html">Extracting</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Analysis</h6>
    <a class="dropdown-item" href="../articles/visualizing.html">Visualizing</a>
    <a class="dropdown-item" href="../articles/expression.html">Expression</a>
    <a class="dropdown-item" href="../articles/enrichment.html">Enrichment</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Workflows</h6>
    <a class="dropdown-item" href="../articles/workflow-simple.html">Simple</a>
    <a class="dropdown-item" href="../articles/workflow-importing.html">Importing Demo</a>
    <a class="dropdown-item" href="../articles/workflow-userimport.html">User Defined Import</a>
    <a class="dropdown-item" href="../articles/workflow-publication.html">Publication Demo</a>
  </div>
</li>
      </ul>
<form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off">
</form>

      <ul class="navbar-nav"></ul>
</div>

    
  </div>
</nav><div class="container template-article">




<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>User Defined Import</h1>
            
      
      
      <div class="d-none name"><code>workflow-userimport.Rmd</code></div>
    </div>

    
    
<p>When working with scientific data, it’s common to come across file
formats that are specific to a particular project, colleague, or
analysis pipeline. These custom file formats can pose a challenge when
trying to import the data into a standard data analysis tool.
Fortunately, tidyproteomics offers a solution by allowing users to
define custom import definitions for these unique file formats.</p>
<p>With tidyproteomics, you can create an import definition that is
tailored to your specific file format. This custom definition will
specify how the data should be read, parsed, and transformed into a tidy
data format. Once the definition is created, you can use it to easily
import data from future files with the same format.</p>
<p>In this document, we will walk you through the process of creating a
custom import definition using tidyproteomics. By the end of this guide,
you’ll have the tools you need to efficiently import data from any
custom file format you may encounter in your scientific work.</p>
<div class="section level2">
<h2 id="an-unknown-dataset">An Unknown Dataset<a class="anchor" aria-label="anchor" href="#an-unknown-dataset"></a>
</h2>
<p>We will be using the data from the PRIDE archive <a href="https://www.ebi.ac.uk/pride/archive/projects/PXD004163" class="external-link">PXD004163</a>,
which fortunately has a decently preserved data file from a TMT
experiment. Reading into the <em>Data Processing Protocol</em> we find a
not-standardized analysis pipeline were raw files were converted to mzML
format, searched using MSGF+ and Percolator, and quantified using OpenMS
project’s IsobaricAnalyzer. PSMs found at 1% FDR were used to infer gene
identities and quantify using PSM quantification ratios.</p>
<p>NOTE: An indepth analysis of this data is provided in the online <a href="https://bioconductor.org/packages/release/bioc/vignettes/DEqMS/inst/doc/DEqMS-package-vignette.html" class="external-link">DEqMS</a>
documentation. The discussion here is simply about how to import data
derived by a method other than those described in the
<code><a href="../articles/importing.html">vignette("importing")</a></code>.</p>
<div class="sourceCode" id="cb1"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">url</span> <span class="op">&lt;-</span> <span class="st">"https://ftp.ebi.ac.uk/pride-archive/2016/06/PXD004163/Yan_miR_Protein_table.flatprottable.txt"</span></span>
<span><span class="fu"><a href="https://rdrr.io/r/utils/download.file.html" class="external-link">download.file</a></span><span class="op">(</span><span class="va">url</span>, destfile <span class="op">=</span> <span class="st">"./miR_Proteintable.tsv"</span>,method <span class="op">=</span> <span class="st">"auto"</span><span class="op">)</span></span></code></pre></div>
<p>After downloading we find the following (transposed here for
readability):</p>
<table style="width:100%;" class="table">
<colgroup>
<col width="16%">
<col width="16%">
<col width="16%">
<col width="16%">
<col width="16%">
<col width="16%">
</colgroup>
<tbody>
<tr class="odd">
<td></td>
<td><strong><em>header</em></strong></td>
<td><strong><em>row_1</em></strong></td>
<td><strong><em>row_2</em></strong></td>
<td><strong><em>row_3</em></strong></td>
<td>…</td>
</tr>
<tr class="even">
<td><strong><em>col_01</em></strong></td>
<td>Protein accession</td>
<td>A2M</td>
<td>A2ML1</td>
<td>AAAS</td>
<td>…</td>
</tr>
<tr class="odd">
<td><strong><em>col_02</em></strong></td>
<td>Gene</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>…</td>
</tr>
<tr class="even">
<td><strong><em>col_03</em></strong></td>
<td>Associated gene ID</td>
<td>ENSG00000175899</td>
<td>ENSG00000166535</td>
<td>ENSG00000094914</td>
<td>…</td>
</tr>
<tr class="odd">
<td><strong><em>col_04</em></strong></td>
<td>Description</td>
<td>alpha-2-macroglobulin [Source:HGNC Symbol;Acc:HGNC:7]</td>
<td>alpha-2-macroglobulin-like 1 [Source:HGNC
Symbol;Acc:HGNC:23336]</td>
<td>achalasia, adrenocortical insufficiency, alacrimia [Source:HGNC</td>
<td>…</td>
</tr>
<tr class="even">
<td><strong><em>col_05</em></strong></td>
<td>Coverage</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>…</td>
</tr>
<tr class="odd">
<td><strong><em>col_06</em></strong></td>
<td># Proteins</td>
<td>na</td>
<td>na</td>
<td>na</td>
<td>…</td>
</tr>
<tr class="even">
<td><strong><em>col_07</em></strong></td>
<td>Proteins in group</td>
<td>NA</td>
<td>NA</td>
<td>NA</td>
<td>…</td>
</tr>
<tr class="odd">
<td><strong><em>col_08</em></strong></td>
<td>miR FASP_# Unique peptides</td>
<td>8</td>
<td>1</td>
<td>6</td>
<td>…</td>
</tr>
<tr class="even">
<td><strong><em>col_09</em></strong></td>
<td>miR FASP_# Peptides</td>
<td>9</td>
<td>1</td>
<td>6</td>
<td>…</td>
</tr>
<tr class="odd">
<td><strong><em>col_10</em></strong></td>
<td>miR FASP_# PSMs</td>
<td>12</td>
<td>1</td>
<td>11</td>
<td>…</td>
</tr>
<tr class="even">
<td><strong><em>col_11</em></strong></td>
<td>miR FASP_Protein error probability</td>
<td>2.87E-38</td>
<td>0.0169213</td>
<td>8.18E-39</td>
<td>…</td>
</tr>
<tr class="odd">
<td><strong><em>col_12</em></strong></td>
<td>miR FASP_q-value</td>
<td>0</td>
<td>0.0141899</td>
<td>0</td>
<td>…</td>
</tr>
<tr class="even">
<td><strong><em>col_13</em></strong></td>
<td>miR FASP_PEP</td>
<td>0.00162245</td>
<td>0.7890065</td>
<td>0.00162245</td>
<td>…</td>
</tr>
<tr class="odd">
<td><strong><em>col_14</em></strong></td>
<td>miR FASP_MS1 precursor area</td>
<td>16352434.3</td>
<td>768887</td>
<td>69955068</td>
<td>…</td>
</tr>
<tr class="even">
<td><strong><em>col_15</em></strong></td>
<td>miR FASP_tmt10plex_126</td>
<td>0.93083081</td>
<td>0.88733015</td>
<td>0.91543699</td>
<td>…</td>
</tr>
<tr class="odd">
<td><strong><em>col_16</em></strong></td>
<td>miR FASP_tmt10plex_126 - # quanted PSMs</td>
<td>11</td>
<td>1</td>
<td>11</td>
<td>…</td>
</tr>
<tr class="even">
<td><strong><em>col_17</em></strong></td>
<td>miR FASP_tmt10plex_127N</td>
<td>0.77202146</td>
<td>1.0947978</td>
<td>0.9168362</td>
<td></td>
</tr>
<tr class="odd">
<td><strong><em>col_18</em></strong></td>
<td>miR FASP_tmt10plex_127N - # quanted PSMs</td>
<td>11</td>
<td>1</td>
<td>11</td>
<td>…</td>
</tr>
<tr class="even">
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
<td>…</td>
</tr>
</tbody>
</table>
<p>From the `vignette(“importing”)` vignette we know that there are 4
must have columns for importing protein quantitative data -
<code>sample: sample</code>, <code>sample: sample_file</code>,
<code>identifier: protein</code>, and
<code>quantitative: abundance_raw</code>.</p>
<p>From looking at the file we can assume:</p>
<ul>
<li><p><code>identifier: protein</code> = <em>Protein accession
(col_01)</em></p></li>
<li><p><code>quantitative: abundance_raw</code> = <em>miR
FASP_tmt10plex_1xx (cols_ 15, 17, 19, 21, 23, 25, 27, 29, 31,
33)</em></p></li>
<li><p><code>sample: sample</code> = will be derived from a pivot on
<code>quantitative: abundance_raw</code></p></li>
<li><p><code>sample: sample_file</code> = not provided, will be assumed
as the import file name</p></li>
</ul>
<p>Additionally we see that some accounting and annotation data is
present:</p>
<ul>
<li><p><code>accounting: num_psms</code> = <em>miR FASP_# PSMs
(col_15)</em></p></li>
<li><p><code>accounting: num_peptides</code> = <em>miR FASP_# Peptides
(col_09)</em></p></li>
<li><p><code>accounting: num_unique_peptides</code> = <em>miR FASP_#
Unique Peptides (col_08)</em></p></li>
<li><p><code>annotation: description</code> = <em>Description
(col_04)</em></p></li>
<li><p><code>annotation: gene_name</code> = <em>Associated gene ID
(col_03)</em></p></li>
</ul>
</div>
<div class="section level2">
<h2 id="build-an-import-definition-file">Build an Import Definition File<a class="anchor" aria-label="anchor" href="#build-an-import-definition-file"></a>
</h2>
<p>We need to construct an import definition file, which has the
objective of pulling in only the needed data, changing the column names
to standard tidyproteomic’s definitions, and pivot the table such that
one row accounts for only one quantitative measure. These definitions
are shown in the `vignette(“importing”)` vignette, and structured by the
following columns:</p>
<ul>
<li><p><code>category</code> - defines which of the 4 SQL-like tables in
the data object this information belongs.</p></li>
<li><p><code>column_defined</code> - defines the SQL-like column name,
using the <a href="https://en.wikipedia.org/wiki/Snake_case" class="external-link">snake_case</a> style
convention.</p></li>
<li><p><code>column_import</code> - a <a href="https://www.regular-expressions.info/" class="external-link">regular expression</a>
defined pattern for recognizing 1 or more columns. This is useful when
each “sample/run” is defined as a column with no exact number of columns
defined for importing.</p></li>
<li><p><code>pattern_extract</code> - a regular expression pattern to
pull in only the data needed, such as extracting just the protein
accession number from a longer text.</p></li>
<li><p><code>pattern_remove</code> - a regular expression pattern to
automatically filter out some results, such as REV_* in protein
accession indicating a decoy.</p></li>
<li><p><code>pattern_split</code> - a string value to split on, for
example when multiple proteins are listed separated by
<code>;</code>.</p></li>
<li><p><code>pivot</code> - a TRUE only indicator that this column
should pivot, such as the quant columns identified above.</p></li>
</ul>
<div class="section level3">
<h3 id="definition-file">Definition File<a class="anchor" aria-label="anchor" href="#definition-file"></a>
</h3>
<p>This is the final defnintion file, saved as a tab-seperated-value
(.tsv) file In our project directory, named
<code>TMTOpenMS_proteins.csv</code>.</p>
<table style="width:100%;" class="table">
<colgroup>
<col width="9%">
<col width="14%">
<col width="28%">
<col width="20%">
<col width="11%">
<col width="10%">
<col width="4%">
</colgroup>
<thead><tr class="header">
<th>category</th>
<th>column_defined</th>
<th>column_import</th>
<th>pattern_extract</th>
<th>pattern_remove</th>
<th>pattern_split</th>
<th>pivot</th>
</tr></thead>
<tbody>
<tr class="odd">
<td>sample</td>
<td>sample</td>
<td></td>
<td>(?&lt;=tmt10plex\_)[0-9]+.+</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>sample</td>
<td>sample_file</td>
<td></td>
<td>.+\_tmt10plex\_[0-9]+.+</td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>identifier</td>
<td>protein</td>
<td>^Protein accession$</td>
<td></td>
<td></td>
<td>\;</td>
<td></td>
</tr>
<tr class="even">
<td>quantitative</td>
<td>abundance_raw</td>
<td>^miR\sFASP\_tmt10plex.{4,5}$</td>
<td></td>
<td></td>
<td></td>
<td>TRUE</td>
</tr>
<tr class="odd">
<td>accounting</td>
<td>num_psms</td>
<td>^miR FASP_# PSMs$</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>accounting</td>
<td>num_peptides</td>
<td>^miR FASP_# Peptides$</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>accounting</td>
<td>num_unique_peptides</td>
<td>^miR FASP_# Unique peptides$</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>accounting</td>
<td>protein_error_prob</td>
<td>^miR FASP_q-value$</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>accounting</td>
<td>protein_qvalue</td>
<td>^miR FASP_Protein error probability$</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="even">
<td>annotation</td>
<td>description</td>
<td>^Description$</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
<tr class="odd">
<td>annotation</td>
<td>gene_name</td>
<td>^Associated gene ID$</td>
<td></td>
<td></td>
<td></td>
<td></td>
</tr>
</tbody>
</table>
</div>
</div>
<div class="section level2">
<h2 id="importing-data">Importing Data<a class="anchor" aria-label="anchor" href="#importing-data"></a>
</h2>
<div class="sourceCode" id="cb2"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://dplyr.tidyverse.org" class="external-link">dplyr</a></span><span class="op">)</span></span>
<span><span class="kw"><a href="https://rdrr.io/r/base/library.html" class="external-link">library</a></span><span class="op">(</span><span class="va"><a href="https://jeffsocal.github.io/tidyproteomics/">tidyproteomics</a></span><span class="op">)</span></span>
<span></span>
<span><span class="va">data_prot</span> <span class="op">&lt;-</span> <span class="st">"miR_Proteintable.tsv"</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/import.html">import</a></span><span class="op">(</span><span class="st">'PXD004163'</span>, <span class="st">'proteins'</span>, path <span class="op">=</span> <span class="st">"TMTOpenMS_proteins.tsv"</span><span class="op">)</span></span></code></pre></div>
<pre><code>ℹ Importing PXD004163:
ℹ ... split protein with \; not detected
ℹ ... no homology detected         
ℹ ... match between runs not found in data
✔ ... parsing miR_Proteintable.tsv [2.1s]
✔ ... tidying the data and finishing up [1.8s]</code></pre>
<p>Printing out the data object we can see we have 10 samples from the
TMT_10plex. We need to reassign the raw TMT labels to some sample
designations and filter the data.</p>
<div class="sourceCode" id="cb4"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_prot</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold; font-style: italic;">Quantitative Proteomics Data Object</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Origin          PXD004163 </span></span>
<span><span class="co">#&gt;                 proteins (15.55 MB) </span></span>
<span><span class="co">#&gt; Composition     10 files </span></span>
<span><span class="co">#&gt;                 10 samples (s126, s127n, s127c, s128n, s128c, s129n, s129c, s130n, s130c, s131) </span></span>
<span><span class="co">#&gt; Quantitation    9638 proteins </span></span>
<span><span class="co">#&gt;                 0.48 log10 dynamic range </span></span>
<span><span class="co">#&gt;                 0.687% missing values </span></span>
<span><span class="co">#&gt;  *imputed        </span></span>
<span><span class="co">#&gt; Accounting      (6) num_unique_peptides num_peptides num_psms protein_qvalue protein_error_prob</span></span>
<span><span class="co">#&gt;                 imputed </span></span>
<span><span class="co">#&gt; Annotations     (2) gene_name description </span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<p>Note that the log10 dynamic range is 0.39, typically for TMT this is
around 3, and for LFQ it can be anywhere from 4 to 8. The plot of
quantitative rank and quantitative value also looks atypical – the
manuscript states:</p>
<div class="line-block">Peptide spectrum matches found at 1% false
discovery rate were used to infer gene identities,<br>
which were quantified using the medians of peptide spectrum matches
quantification ratios.</div>
<div class="sourceCode" id="cb5"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_prot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_quantrank.html">plot_quantrank</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="workflow-userimport_files/figure-html/unnamed-chunk-6-1.png" width="700"></p>
<p>The TMT labeling assignments are not not provided in the literature,
but can be found in the DEqMS tutorial:</p>
<div class="sourceCode" id="cb6"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_prot</span> <span class="op">&lt;-</span> <span class="va">data_prot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/subset.html">subset</a></span><span class="op">(</span><span class="va">protein_qvalue</span> <span class="op">&lt;</span> <span class="fl">0.01</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/reassign.html">reassign</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'s126'</span>, .replace <span class="op">=</span> <span class="st">'ctrl'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/reassign.html">reassign</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'s127n'</span>, .replace <span class="op">=</span> <span class="st">'miR191'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/reassign.html">reassign</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'s127c'</span>, .replace <span class="op">=</span> <span class="st">'miR372'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/reassign.html">reassign</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'s128n'</span>, .replace <span class="op">=</span> <span class="st">'miR519'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/reassign.html">reassign</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'s128c'</span>, .replace <span class="op">=</span> <span class="st">'ctrl'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/reassign.html">reassign</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'s129n'</span>, .replace <span class="op">=</span> <span class="st">'miR372'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/reassign.html">reassign</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'s129c'</span>, .replace <span class="op">=</span> <span class="st">'miR519'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/reassign.html">reassign</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'s130n'</span>, .replace <span class="op">=</span> <span class="st">'ctrl'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/reassign.html">reassign</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'s130c'</span>, .replace <span class="op">=</span> <span class="st">'miR191'</span><span class="op">)</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span></span>
<span>  <span class="fu"><a href="../reference/reassign.html">reassign</a></span><span class="op">(</span><span class="va">sample</span> <span class="op">==</span> <span class="st">'s131'</span>, .replace <span class="op">=</span> <span class="st">'miR372'</span><span class="op">)</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Subsetting data: <span style="color: #FF5F00; font-style: italic;">protein_qvalue</span> <span style="color: #0000BB;">&lt;</span> <span style="color: #FF5F00; font-style: italic;">0.01</span></span></span>
<span><span class="co">#&gt; <span style="color: #00BB00;">✔</span> Subsetting data: <span style="color: #FF5F00; font-style: italic;">protein_qvalue</span> <span style="color: #0000BB;">&lt;</span> <span style="color: #FF5F00; font-style: italic;">0.01</span> ... done</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Reassigning 1</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Reassigning 2</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Reassigning 3</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Reassigning 4</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Reassigning 5</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Reassigning 6</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Reassigning 7</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Reassigning 8</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Reassigning 9</span></span>
<span><span class="co">#&gt; <span style="color: #00BBBB;">ℹ</span> Reassigning 10</span></span>
<span></span>
<span><span class="va">data_prot</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; ── <span style="font-weight: bold; font-style: italic;">Quantitative Proteomics Data Object</span> ──</span></span>
<span><span class="co">#&gt; </span></span>
<span><span class="co">#&gt; Origin          PXD004163 </span></span>
<span><span class="co">#&gt;                 proteins (14.40 MB) </span></span>
<span><span class="co">#&gt; Composition     10 files </span></span>
<span><span class="co">#&gt;                 4 samples (ctrl, miR191, miR372, miR519) </span></span>
<span><span class="co">#&gt; Quantitation    8742 proteins </span></span>
<span><span class="co">#&gt;                 0.39 log10 dynamic range </span></span>
<span><span class="co">#&gt;                 0.216% missing values </span></span>
<span><span class="co">#&gt;  *imputed        </span></span>
<span><span class="co">#&gt; Accounting      (6) num_unique_peptides num_peptides num_psms protein_qvalue protein_error_prob</span></span>
<span><span class="co">#&gt;                 imputed </span></span>
<span><span class="co">#&gt; Annotations     (2) gene_name description </span></span>
<span><span class="co">#&gt; </span></span></code></pre></div>
<div class="sourceCode" id="cb7"><pre class="downlit sourceCode r">
<code class="sourceCode R"><span><span class="va">data_prot</span> <span class="op"><a href="https://magrittr.tidyverse.org/reference/pipe.html" class="external-link">%&gt;%</a></span> <span class="fu"><a href="../reference/plot_pca.html">plot_pca</a></span><span class="op">(</span><span class="op">)</span></span></code></pre></div>
<p><img src="workflow-userimport_files/figure-html/unnamed-chunk-8-1.png" width="700"></p>
<p>Note that DEqMS <em>“<strong>.. takes into account the inherent
dependence of protein variance on the number of PSMs or peptides used
for quantification, thereby providing a more accurate variance
estimation</strong>.”</em> and as such PSM count need to be be
integrated into this analysis, following the protocol outlined in the
documentation. The purpose of this vignette is not to replicate the
DEqMS method, rather to demonstrate how one would build an import
definition file to utilize tidyproteomics.</p>
</div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside>
</div>



    <footer><div class="pkgdown-footer-left">
  <p></p>
<p>Developed by Jeff Jones.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p>
<p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.7.</p>
</div>

    </footer>
</div>

  

  

  </body>
</html>
