<!DOCTYPE html>
<!-- Generated by pkgdown: do not edit by hand --><html lang="en"><head><meta http-equiv="Content-Type" content="text/html; charset=UTF-8"><meta charset="utf-8"><meta http-equiv="X-UA-Compatible" content="IE=edge"><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><title>Function reference â€¢ tidyproteomics</title><!-- favicons --><link rel="icon" type="image/png" sizes="16x16" href="../favicon-16x16.png"><link rel="icon" type="image/png" sizes="32x32" href="../favicon-32x32.png"><link rel="apple-touch-icon" type="image/png" sizes="180x180" href="../apple-touch-icon.png"><link rel="apple-touch-icon" type="image/png" sizes="120x120" href="../apple-touch-icon-120x120.png"><link rel="apple-touch-icon" type="image/png" sizes="76x76" href="../apple-touch-icon-76x76.png"><link rel="apple-touch-icon" type="image/png" sizes="60x60" href="../apple-touch-icon-60x60.png"><script src="../deps/jquery-3.6.0/jquery-3.6.0.min.js"></script><meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no"><link href="../deps/bootstrap-5.1.3/bootstrap.min.css" rel="stylesheet"><script src="../deps/bootstrap-5.1.3/bootstrap.bundle.min.js"></script><!-- Font Awesome icons --><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/all.min.css" integrity="sha256-mmgLkCYLUQbXn0B1SRqzHar6dCnv9oZFPEC1g1cwlkk=" crossorigin="anonymous"><link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.12.1/css/v4-shims.min.css" integrity="sha256-wZjR52fzng1pJHwx4aV2AO3yyTOXrcDW7jBpJtTwVxw=" crossorigin="anonymous"><!-- bootstrap-toc --><script src="https://cdn.rawgit.com/afeld/bootstrap-toc/v1.0.1/dist/bootstrap-toc.min.js"></script><!-- headroom.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/headroom.min.js" integrity="sha256-AsUX4SJE1+yuDu5+mAVzJbuYNPHj/WroHuZ8Ir/CkE0=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/headroom/0.11.0/jQuery.headroom.min.js" integrity="sha256-ZX/yNShbjqsohH1k95liqY9Gd8uOiE1S4vZc+9KQ1K4=" crossorigin="anonymous"></script><!-- clipboard.js --><script src="https://cdnjs.cloudflare.com/ajax/libs/clipboard.js/2.0.6/clipboard.min.js" integrity="sha256-inc5kl9MA1hkeYUt+EC3BhlIgyp/2jDIyBLS6k3UxPI=" crossorigin="anonymous"></script><!-- search --><script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/6.4.6/fuse.js" integrity="sha512-zv6Ywkjyktsohkbp9bb45V6tEMoWhzFzXis+LrMehmJZZSys19Yxf1dopHx7WzIKxr5tK2dVcYmaCk2uqdjF4A==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/autocomplete.js/0.38.0/autocomplete.jquery.min.js" integrity="sha512-GU9ayf+66Xx2TmpxqJpliWbT5PiGYxpaG8rfnBEk1LL8l1KGkRShhngwdXK1UgqhAzWpZHSiYPc09/NwDQIGyg==" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/mark.min.js" integrity="sha512-5CYOlHXGh6QpOFA/TeTylKLWfB3ftPsde7AnmhuitiTX4K5SqCLBeKro6sPS8ilsz1Q4NRx3v8Ko2IBiszzdww==" crossorigin="anonymous"></script><!-- pkgdown --><script src="../pkgdown.js"></script><meta property="og:title" content="Function reference"><meta property="og:image" content="https://jeffsocal.github.io/tidyproteomics/logo.png"><!-- mathjax --><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/MathJax.js" integrity="sha256-nvJJv9wWKEm88qvoQl9ekL2J+k/RWIsaSScxxlsrv8k=" crossorigin="anonymous"></script><script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.5/config/TeX-AMS-MML_HTMLorMML.js" integrity="sha256-84DKXVJXs0/F8OTMzX4UR909+jtl4G7SPypPavF+GfA=" crossorigin="anonymous"></script><!--[if lt IE 9]>
<script src="https://oss.maxcdn.com/html5shiv/3.7.3/html5shiv.min.js"></script>
<script src="https://oss.maxcdn.com/respond/1.4.2/respond.min.js"></script>
<![endif]--></head><body>
    <a href="#main" class="visually-hidden-focusable">Skip to contents</a>
    

    <nav class="navbar fixed-top navbar-light navbar-expand-lg bg-light"><div class="container">
    
    <a class="navbar-brand me-2" href="../index.html">tidyproteomics</a>

    <small class="nav-text text-muted me-auto" data-bs-toggle="tooltip" data-bs-placement="bottom" title="">1.4.2</small>

    
    <button class="navbar-toggler" type="button" data-bs-toggle="collapse" data-bs-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
      <span class="navbar-toggler-icon"></span>
    </button>

    <div id="navbar" class="collapse navbar-collapse ms-3">
      <ul class="navbar-nav me-auto"><li class="nav-item">
  <a class="nav-link" href="../articles/overview.html">Overview</a>
</li>
<li class="active nav-item">
  <a class="nav-link" href="../reference/index.html">Reference</a>
</li>
<li class="nav-item dropdown">
  <a href="#" class="nav-link dropdown-toggle" data-bs-toggle="dropdown" role="button" aria-expanded="false" aria-haspopup="true" id="dropdown-articles">Articles</a>
  <div class="dropdown-menu" aria-labelledby="dropdown-articles">
    <a class="dropdown-item" href="../articles/updates.html">Updates</a>
    <a class="dropdown-item" href="../articles/data-object.html">The Data-Object</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Operations</h6>
    <a class="dropdown-item" href="../articles/importing.html">Importing</a>
    <a class="dropdown-item" href="../articles/summarizing.html">Summarizing</a>
    <a class="dropdown-item" href="../articles/subsetting.html">Subsetting</a>
    <a class="dropdown-item" href="../articles/normalizing.html">Normalizing</a>
    <a class="dropdown-item" href="../articles/imputing.html">Imputing</a>
    <a class="dropdown-item" href="../articles/annotating.html">Annotating</a>
    <a class="dropdown-item" href="../articles/collapsing.html">Protein Accounting</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Analysis</h6>
    <a class="dropdown-item" href="../articles/visualizing.html">Visualizing</a>
    <a class="dropdown-item" href="../articles/expression.html">Expression</a>
    <a class="dropdown-item" href="../articles/enrichment.html">Enrichment</a>
    <div class="dropdown-divider"></div>
    <h6 class="dropdown-header" data-toc-skip>Workflows</h6>
    <a class="dropdown-item" href="../articles/workflow-simple.html">Simple</a>
    <a class="dropdown-item" href="../articles/workflow-importing.html">Importing Demo</a>
    <a class="dropdown-item" href="../articles/workflow-userimport.html">User Defined Import</a>
    <a class="dropdown-item" href="../articles/workflow-publication.html">Publication Demo</a>
  </div>
</li>
      </ul><form class="form-inline my-2 my-lg-0" role="search">
        <input type="search" class="form-control me-sm-2" aria-label="Toggle navigation" name="search-input" data-search-index="../search.json" id="search-input" placeholder="Search for" autocomplete="off"></form>

      <ul class="navbar-nav"></ul></div>

    
  </div>
</nav><div class="container template-reference-index">
<div class="row">
  <main id="main" class="col-md-9"><div class="page-header">
      <img src="../logo.png" class="logo" alt=""><h1>Function reference</h1>
    </div>

    <div class="section level2">
      <h2 id="import-export">Import / Export<a class="anchor" aria-label="anchor" href="#import-export"></a></h2>
      
      <p class="section-desc"></p><p>Data can imported from flat tables or file directories, and exported again into flat tables. See <code><a href="../articles/importing.html">vignette("importing")</a></code> for more details and examples.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="import.html">import()</a></code> 
        </dt>
        <dd>Main function for importing data to the tidyproteomics list object</dd>
      </dl><dl><dt>
          
          <code><a href="export_quant.html">export_quant()</a></code> 
        </dt>
        <dd>Export the quantitative data from an tidyproteomics data-object</dd>
      </dl><dl><dt>
          
          <code><a href="export_analysis.html">export_analysis()</a></code> 
        </dt>
        <dd>Export the quantitative data from an tidyproteomics data-object</dd>
      </dl><dl><dt>
          
          <code><a href="as.data.frame.tidyproteomics.html">as.data.frame(<i>&lt;tidyproteomics&gt;</i>)</a></code> 
        </dt>
        <dd>Helper function to convert the data-object into a tibble</dd>
      </dl><dl><dt>
          
          <code><a href="load_local.html">load_local()</a></code> 
        </dt>
        <dd>Load project specific data</dd>
      </dl><dl><dt>
          
          <code><a href="save_local.html">save_local()</a></code> 
        </dt>
        <dd>Store data locally</dd>
      </dl><dl><dt>
          
          <code><a href="save_table.html">save_table()</a></code> 
        </dt>
        <dd>Write table data locally</dd>
      </dl></div><div class="section level2">
      <h2 id="helper-functions">Helper functions<a class="anchor" aria-label="anchor" href="#helper-functions"></a></h2>
      
      <p class="section-desc"></p><p>These functions can save data and tables out to the project folder, add annotations to the data and tables, modify the experimental names, and list out the sequence of data transformations.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="merge.html">merge()</a></code> 
        </dt>
        <dd>Merge multiple tidyproteomics data-objects</dd>
      </dl><dl><dt>
          
          <code><a href="collapse.html">collapse()</a></code> 
        </dt>
        <dd>Convert peptide quantitative data into protein quantitative data</dd>
      </dl><dl><dt>
          
          <code><a href="operations.html">operations()</a></code> 
        </dt>
        <dd>Returns the data transformations</dd>
      </dl><dl><dt>
          
          <code><a href="reassign.html">reassign()</a></code> 
        </dt>
        <dd>reassign the sample info</dd>
      </dl><dl><dt>
          
          <code><a href="annotate.html">annotate()</a></code> 
        </dt>
        <dd>Main function for adding annotations to a tidyproteomics data-object</dd>
      </dl><dl><dt>
          
          <code><a href="show_annotations.html">show_annotations()</a></code> 
        </dt>
        <dd>Display the current annotation data</dd>
      </dl><dl><dt>
          
          <code><a href="path_to_package_data.html">path_to_package_data()</a></code> 
        </dt>
        <dd>Helper function for displaying path to data</dd>
      </dl></div><div class="section level2">
      <h2 id="subsetting">Subsetting<a class="anchor" aria-label="anchor" href="#subsetting"></a></h2>
      
      <p class="section-desc"></p><p><strong>Subsetting</strong> filters an tidyproteomics data-object by a given regular expression to create a subset. See <code><a href="../articles/subsetting.html">vignette("subsetting")</a></code> for more details and examples.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="subset.html">subset()</a></code> 
        </dt>
        <dd>Create a data subset</dd>
      </dl></div><div class="section level2">
      <h2 id="summarizing">Summarizing<a class="anchor" aria-label="anchor" href="#summarizing"></a></h2>
      
      <p class="section-desc"></p><p><strong>Summarizing</strong> computes quick summary statistics on a given tidyproteomics data-object. See <code>vignette("summary")</code> for more details and examples.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="print.tidyproteomics.html">print(<i>&lt;tidyproteomics&gt;</i>)</a></code> 
        </dt>
        <dd>Tidy-Quant data object print definition</dd>
      </dl><dl><dt>
          
          <code><a href="plot.tidyproteomics.html">plot(<i>&lt;tidyproteomics&gt;</i>)</a></code> 
        </dt>
        <dd>Tidy-Quant data object plot definition</dd>
      </dl><dl><dt>
          
          <code><a href="summary.tidyproteomics.html">summary(<i>&lt;tidyproteomics&gt;</i>)</a></code> 
        </dt>
        <dd>Summarize the data</dd>
      </dl><dl><dt>
          
          <code><a href="plot_counts.html">plot_counts()</a></code> 
        </dt>
        <dd>Plot the accounting of proteins. peptides, and other counts</dd>
      </dl><dl><dt>
          
          <code><a href="plot_quantrank.html">plot_quantrank()</a></code> 
        </dt>
        <dd>Plot the variation in normalized values</dd>
      </dl></div><div class="section level2">
      <h2 id="normalizing">Normalizing<a class="anchor" aria-label="anchor" href="#normalizing"></a></h2>
      
      <p class="section-desc"></p><p><strong>Normalizing</strong> adjusts the quantitiative values between samples to remove collection biases, based on a number of methods. See <code><a href="../articles/normalizing.html">vignette("normalizing")</a></code> for more details and examples.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="normalize.html">normalize()</a></code> 
        </dt>
        <dd>Main function for normalizing quantitative data in a tidyproteomics data-object</dd>
      </dl><dl><dt>
          
          <code><a href="select_normalization.html">select_normalization()</a></code> 
        </dt>
        <dd>Select a normalization method</dd>
      </dl></div><div class="section level2">
      <h2 id="imputing">Imputing<a class="anchor" aria-label="anchor" href="#imputing"></a></h2>
      
      <p class="section-desc"></p><p><strong>Imputing</strong> calculates values for missing observations, based on a number of methods. See <code><a href="../articles/imputing.html">vignette("imputing")</a></code> for more details and examples.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="impute.html">impute()</a></code> 
        </dt>
        <dd>Main method for imputing missing values</dd>
      </dl></div><div class="section level2">
      <h2 id="visualization">Visualization<a class="anchor" aria-label="anchor" href="#visualization"></a></h2>
      
      <p class="section-desc"></p><p><strong>Visualizing</strong> data is important for understanding the underlying quality of the data as well as generating graphical representations for documenting work. See <code>vignette("visualization")</code> for more details and examples.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="plot_normalization.html">plot_normalization()</a></code> 
        </dt>
        <dd>Plot normalized values</dd>
      </dl><dl><dt>
          
          <code><a href="plot_variation_cv.html">plot_variation_cv()</a></code> 
        </dt>
        <dd>Plot the variation in normalized values</dd>
      </dl><dl><dt>
          
          <code><a href="plot_variation_pca.html">plot_variation_pca()</a></code> 
        </dt>
        <dd>Plot the PCA variation in normalized values</dd>
      </dl><dl><dt>
          
          <code><a href="plot_dynamic_range.html">plot_dynamic_range()</a></code> 
        </dt>
        <dd>Plot CVs by abundance</dd>
      </dl><dl><dt>
          
          <code><a href="plot_venn.html">plot_venn()</a></code> 
        </dt>
        <dd>GGplot2 extension to plot a Venn diagram</dd>
      </dl><dl><dt>
          
          <code><a href="plot_euler.html">plot_euler()</a></code> 
        </dt>
        <dd>GGplot2 extension to plot a Euler diagram</dd>
      </dl><dl><dt>
          
          <code><a href="plot_pca.html">plot_pca()</a></code> 
        </dt>
        <dd>Plot PCA values</dd>
      </dl><dl><dt>
          
          <code><a href="plot_heatmap.html">plot_heatmap()</a></code> 
        </dt>
        <dd>Plot a heatmap of quantitative values by sample</dd>
      </dl></div><div class="section level2">
      <h2 id="two-sample-analysis">Two-sample Analysis<a class="anchor" aria-label="anchor" href="#two-sample-analysis"></a></h2>
      
      <p class="section-desc"></p><p><strong>Two-sample Analysis</strong> is a process by which two samples are statistically compared to identify potential observational outliers worth explroing further. See <code><a href="../articles/expression.html">vignette("expression")</a></code> and <code><a href="../articles/enrichment.html">vignette("enrichment")</a></code> for more details and examples.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="expression.html">expression()</a></code> 
        </dt>
        <dd>Summarize the data</dd>
      </dl><dl><dt>
          
          <code><a href="plot_volcano.html">plot_volcano()</a></code> 
        </dt>
        <dd>Volcano plot of expression values</dd>
      </dl><dl><dt>
          
          <code><a href="plot_proportion.html">plot_proportion()</a></code> 
        </dt>
        <dd>Plot proportional expression values</dd>
      </dl><dl><dt>
          
          <code><a href="plot_compexp.html">plot_compexp()</a></code> 
        </dt>
        <dd>Comparative analysis between two expression tests</dd>
      </dl><dl><dt>
          
          <code><a href="enrichment.html">enrichment()</a></code> 
        </dt>
        <dd>Compute protein enrichment</dd>
      </dl><dl><dt>
          
          <code><a href="plot_enrichment.html">plot_enrichment()</a></code> 
        </dt>
        <dd>Bubble plot of enrichment values</dd>
      </dl></div><div class="section level2">
      <h2 id="protein-sequencing">Protein Sequencing<a class="anchor" aria-label="anchor" href="#protein-sequencing"></a></h2>
      
      <p class="section-desc"></p><p><strong>Protein Sequencing</strong> allows for the visulaization of peptide sequences and modifications mapped back onto the full protein sequence. See <code>vignette("sequencing")</code> for more details and examples.</p>

      
    </div><div id="" class="section level2">
      
      
      

      <dl><dt>
          
          <code><a href="plot_protein.html">plot_protein()</a></code> 
        </dt>
        <dd>Visualize mapped sequence data</dd>
      </dl><dl><dt>
          
          <code><a href="protein_map.html">protein_map()</a></code> 
        </dt>
        <dd>Align a peptide data to protein sequences for visualization</dd>
      </dl></div>
  </main><aside class="col-md-3"><nav id="toc"><h2>On this page</h2>
    </nav></aside></div>


    <footer><div class="pkgdown-footer-left">
  <p></p><p>Developed by Jeff Jones.</p>
</div>

<div class="pkgdown-footer-right">
  <p></p><p>Site built with <a href="https://pkgdown.r-lib.org/" class="external-link">pkgdown</a> 2.0.6.</p>
</div>

    </footer></div>

  

  

  </body></html>

